deploy/static/provider/do/deploy.yaml

```diff
@@ -441,6 +441,7 @@ webhooks:
         namespace: ingress-nginx
         name: ingress-nginx-controller-admission
         path: /networking/v1/ingresses
+    timeoutSeconds: 29
```

Comment:
I think we can add this to Helm chart's values file.

Reply:
I see, I had a suspicion that these files were generated but I'm not familiar with helm. It will take too much time for me to do it, but I don't mind to contribute / donate if someone is willing to pick this up.

Reply:
I found it!

https://github.com/kubernetes/ingress-nginx/pull/7319/commits/7ffc2a22a7c7e5d08ce787c62291cf7602d09834

I would be grateful for any help moving this through testing.

Reply:
👍 if you need any help please feel free to ping me

Reply:
well, i'm giving it a go after all.

I managed to install go, helm, kind and make on Ubuntu WSL2, but while running `make dev-env`, it cannot connect to `localhost:8080` due to a limitation in WSL2 where the ubuntu instance cannot access the host machine under the name 'localhost'.

Is there anywhere where I can replace the hostname which is used during `make dev-env`? even if its temporarily/locally?

Reply:
I overwrote localhost with my windows IP in `/etc/hosts` , but I still get the same error:

```
Image: "gcr.io/k8s-staging-ingress-nginx/controller:1.0.0-dev" with ID "sha256:dbe09757f6c014bb253f8f45dd43238a8c37fc2f2bc326d40f1e07ffc1272341" not yet present on node "ingress-nginx-dev-control-plane", loading...
[dev-env] deploying NGINX Ingress controller...
Unable to connect to the server: dial tcp 192.168.0.157:8080: i/o timeout
make: *** [Makefile:153: dev-env] Error 1
```

I confirmed that neither in Windows nor in Ubuntu any service is running on 8080 (`telnet 127.0.0.1 8080` on both envs).

Any idea what might go wrong?

Reply:
do you set any proxy before you run `make dev-env`?

Reply:
I have no proxies configured in my WSL, no.

Reply:
oh, now it seems to work...

i deleted `ingress-nginx-dev-control-plane` and re-ran `make dev-env` :-S

Reply:
Okay, make test failed, but make e2e passed.

Not sure what to do with the normal make tests.

Maybe its good enough for further testing by you guys?

<details>
<summary>make test Failed</summary>

```
go: downloading k8s.io/client-go v0.21.1
go: downloading golang.org/x/crypto v0.0.0-20210220033148-5ea612d1eb83
go: downloading k8s.io/klog/v2 v2.8.0
go: downloading github.com/stretchr/testify v1.6.1
go: downloading github.com/go-logr/logr v0.4.0
go: downloading k8s.io/apimachinery v0.21.1
go: downloading github.com/eapache/channels v1.1.0
go: downloading github.com/onsi/ginkgo v1.16.4
go: downloading github.com/armon/go-proxyproto v0.0.0-20200108142055-f0b8253b1507
go: downloading k8s.io/api v0.21.1
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/mitchellh/mapstructure v1.3.2
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading google.golang.org/grpc v1.27.1
go: downloading github.com/mitchellh/go-ps v1.0.0
go: downloading github.com/spf13/cobra v1.1.1
go: downloading github.com/ncabatoff/process-exporter v0.7.2
go: downloading github.com/prometheus/common v0.14.0
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/prometheus/client_golang v1.7.1
go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
go: downloading github.com/google/gofuzz v1.1.0
go: downloading k8s.io/utils v0.0.0-20201110183641-67b214c5f920
go: downloading k8s.io/cli-runtime v0.21.1
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/opencontainers/runc v1.0.0-rc92
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading github.com/eapache/queue v1.1.0
go: downloading sigs.k8s.io/structured-merge-diff/v4 v4.1.0
go: downloading golang.org/x/net v0.0.0-20210224082022-3d97a244fca7
go: downloading github.com/mitchellh/hashstructure v1.0.0
go: downloading golang.org/x/sys v0.0.0-20210225134936-a50acf3fe073
go: downloading github.com/json-iterator/go v1.1.10
go: downloading github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
go: downloading github.com/mitchellh/copystructure v1.0.0
go: downloading github.com/docker/go-units v0.4.0
go: downloading github.com/zakjan/cert-chain-resolver v0.0.0-20200729110141-6b99e360f97a
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading github.com/hashicorp/golang-lru v0.5.4
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369
go: downloading k8s.io/apiextensions-apiserver v0.21.1
go: downloading github.com/mitchellh/reflectwalk v1.0.0
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/sirupsen/logrus v1.7.0
go: downloading sigs.k8s.io/yaml v1.2.0
go: downloading golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/nxadm/tail v1.4.8
go: downloading github.com/fsnotify/fsnotify v1.4.9
go: downloading github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e
go: downloading sigs.k8s.io/kustomize/api v0.8.8
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/google/go-cmp v0.5.2
go: downloading github.com/cyphar/filepath-securejoin v0.2.2
go: downloading gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading golang.org/x/text v0.3.4
go: downloading github.com/moul/pb v0.0.0-20180404114147-54bdd96e6a52
go: downloading cloud.google.com/go v0.54.0
go: downloading google.golang.org/genproto v0.0.0-20201110150050-8816d57aaa9a
go: downloading github.com/coreos/go-systemd/v22 v22.1.0
go: downloading github.com/cespare/xxhash v1.1.0
go: downloading github.com/spf13/pflag v1.0.5
go: downloading github.com/imdario/mergo v0.3.10
go: downloading gopkg.in/go-playground/pool.v3 v3.1.1
go: downloading github.com/peterbourgon/diskv v2.0.1+incompatible
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/golang/protobuf v1.4.3
go: downloading sigs.k8s.io/kustomize/kyaml v0.10.17
go: downloading github.com/gregjones/httpcache v0.0.0-20180305231024-9cad4c3443a7
go: downloading github.com/googleapis/gnostic v0.5.1
go: downloading k8s.io/component-base v0.21.1
go: downloading github.com/Azure/go-autorest v14.2.0+incompatible
go: downloading github.com/fullsailor/pkcs7 v0.0.0-20190404230743-d7302db945fa
go: downloading pault.ag/go/sniff v0.0.0-20200207005214-cf7e4d167732
go: downloading github.com/Azure/go-autorest/autorest v0.11.12
go: downloading google.golang.org/protobuf v1.25.0
go: downloading github.com/evanphx/json-patch v4.9.0+incompatible
go: downloading github.com/ncabatoff/go-seq v0.0.0-20180805175032-b08ef85ed833
go: downloading k8s.io/apiserver v0.21.1
go: downloading github.com/gavv/httpexpect/v2 v2.1.0
go: downloading github.com/Azure/go-autorest/autorest/adal v0.9.5
go: downloading golang.org/x/term v0.0.0-20210220032956-6a3ed077a48d
go: downloading github.com/go-openapi/spec v0.19.5
go: downloading github.com/godbus/dbus/v5 v5.0.3
go: downloading github.com/form3tech-oss/jwt-go v3.2.2+incompatible
go: downloading github.com/go-openapi/jsonpointer v0.19.3
go: downloading github.com/prometheus/procfs v0.2.0
go: downloading golang.org/x/time v0.0.0-20210220033141-f8bda1e9f3ba
go: downloading github.com/xlab/treeprint v0.0.0-20181112141820-a009c3971eca
go: downloading github.com/Azure/go-autorest/autorest/date v0.3.0
go: downloading github.com/imkira/go-interpol v1.0.0
go: downloading github.com/gorilla/websocket v1.4.2
go: downloading github.com/valyala/fasthttp v1.9.0
go: downloading github.com/ajg/form v1.5.1
go: downloading github.com/go-openapi/swag v0.19.5
go: downloading github.com/go-openapi/jsonreference v0.19.3
go: downloading github.com/monochromegane/go-gitignore v0.0.0-20200626010858-205db1a8cc00
go: downloading moul.io/http2curl v1.0.1-0.20190925090545-5cd742060b0e
go: downloading github.com/PuerkitoBio/purell v1.1.1
go: downloading github.com/google/uuid v1.1.2
go: downloading github.com/mailru/easyjson v0.7.0
go: downloading github.com/yudai/gojsondiff v1.0.0
go: downloading github.com/Azure/go-autorest/logger v0.2.0
go: downloading github.com/fatih/structs v1.0.0
go: downloading github.com/klauspost/compress v1.8.2
go: downloading github.com/sergi/go-diff v1.1.0
go: downloading github.com/yalp/jsonpath v0.0.0-20180802001716-5cc68e5049a0
go: downloading github.com/yudai/golcs v0.0.0-20170316035057-ecda9a501e82
go: downloading github.com/google/go-querystring v1.0.0
go: downloading go.uber.org/zap v1.15.0
go: downloading github.com/Azure/go-autorest/tracing v0.6.0
go: downloading github.com/xeipuuv/gojsonschema v1.1.0
go: downloading github.com/emicklei/go-restful v2.9.5+incompatible
go: downloading k8s.io/kube-openapi v0.0.0-20210305001622-591a79e4bda7
go: downloading github.com/liggitt/tabwriter v0.0.0-20181228230101-89fcab3d43de
go: downloading go.uber.org/atomic v1.6.0
go: downloading github.com/valyala/bytebufferpool v1.0.0
go: downloading github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
go: downloading github.com/go-errors/errors v1.0.1
go: downloading github.com/google/btree v1.0.0
go: downloading github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578
go: downloading github.com/xeipuuv/gojsonreference v0.0.0-20180127040603-bd5ef7bd5415
go: downloading github.com/blang/semver v3.5.1+incompatible
go: downloading go.uber.org/multierr v1.5.0
go: downloading github.com/opencontainers/runtime-spec v1.0.3-0.20200728170252-4d89ac9fbff6
go: downloading go.starlark.net v0.0.0-20200306205701-8dd3e2ee1dd5
go: downloading github.com/xeipuuv/gojsonpointer v0.0.0-20180127040702-4e3ac2762d5f
go: downloading github.com/klauspost/cpuid v1.2.1
go: downloading sigs.k8s.io/controller-runtime v0.8.0
go: downloading github.com/kylelemons/godebug v1.1.0
go: downloading github.com/onsi/gomega v1.10.2
go: downloading golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
?       k8s.io/ingress-nginx/cmd/dbg    [no test files]
=== RUN   TestNoMandatoryFlag
--- PASS: TestNoMandatoryFlag (0.00s)
=== RUN   TestDefaults
--- PASS: TestDefaults (0.00s)
=== RUN   TestSetupSSLProxy
--- PASS: TestSetupSSLProxy (0.00s)
=== RUN   TestFlagConflict
--- PASS: TestFlagConflict (0.00s)
=== RUN   TestMaxmindEdition
--- PASS: TestMaxmindEdition (0.00s)
=== RUN   TestMaxmindMirror
--- PASS: TestMaxmindMirror (0.00s)
=== RUN   TestCreateApiserverClient
W0704 14:30:08.179737    1924 client_config.go:615] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
W0704 14:30:08.179763    1924 client_config.go:620] error creating inClusterConfig, falling back to default config: unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined
--- PASS: TestCreateApiserverClient (0.00s)
=== RUN   TestHandleSigterm
W0704 14:30:08.180879    1924 store.go:620] Unexpected error reading configuration configmap: configmaps "" not found
    main_test.go:116: Sending SIGTERM to PID 1924
--- PASS: TestHandleSigterm (1.01s)
PASS
ok      k8s.io/ingress-nginx/cmd/nginx  1.027s
?       k8s.io/ingress-nginx/cmd/plugin [no test files]
?       k8s.io/ingress-nginx/cmd/plugin/commands/backends       [no test files]
?       k8s.io/ingress-nginx/cmd/plugin/commands/certs  [no test files]
?       k8s.io/ingress-nginx/cmd/plugin/commands/conf   [no test files]
?       k8s.io/ingress-nginx/cmd/plugin/commands/exec   [no test files]
?       k8s.io/ingress-nginx/cmd/plugin/commands/general        [no test files]
?       k8s.io/ingress-nginx/cmd/plugin/commands/info   [no test files]
?       k8s.io/ingress-nginx/cmd/plugin/commands/ingresses      [no test files]
?       k8s.io/ingress-nginx/cmd/plugin/commands/lint   [no test files]
?       k8s.io/ingress-nginx/cmd/plugin/commands/logs   [no test files]
?       k8s.io/ingress-nginx/cmd/plugin/commands/ssh    [no test files]
?       k8s.io/ingress-nginx/cmd/plugin/kubectl [no test files]
?       k8s.io/ingress-nginx/cmd/plugin/lints   [no test files]
?       k8s.io/ingress-nginx/cmd/plugin/request [no test files]
?       k8s.io/ingress-nginx/cmd/plugin/util    [no test files]
?       k8s.io/ingress-nginx/cmd/waitshutdown   [no test files]
=== RUN   TestHandleAdmission
E0704 13:13:26.430717   15474 main.go:78] "failed to decode ingress" err="couldn't get version/kind; json parse error: invalid character 'ÿ' looking for beginning of value"
E0704 13:13:26.432185   15474 main.go:90] "invalid ingress configuration" err="this is a test error" ingress="/"
I0704 13:13:26.432249   15474 main.go:101] "successfully validated configuration, accepting" ingress="/"
--- PASS: TestHandleAdmission (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/admission/controller      (cached)
=== RUN   TestSHA1
E0704 13:13:20.808434    7275 file.go:32] "Error reading file" err="open : no such file or directory" path=""
--- PASS: TestSHA1 (0.01s)
PASS
ok      k8s.io/ingress-nginx/internal/file      (cached)
=== RUN   TestEqualConfiguration
--- PASS: TestEqualConfiguration (0.01s)
=== RUN   TestL4ServiceElementsMatch
--- PASS: TestL4ServiceElementsMatch (0.00s)
=== RUN   TestIntElementsMatch
--- PASS: TestIntElementsMatch (0.00s)
=== RUN   TestSSLCertMatch
--- PASS: TestSSLCertMatch (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress   (cached)
=== RUN   TestSSLPassthrough
--- PASS: TestSSLPassthrough (0.05s)
=== RUN   TestUpstreamHashBy
--- PASS: TestUpstreamHashBy (0.00s)
=== RUN   TestAffinitySession
    annotations_test.go:187: Testing pass cookie route
    annotations_test.go:187: Testing pass cookie route1
    annotations_test.go:187: Testing pass cookie INGRESSCOOKIE
    annotations_test.go:187: Testing pass
    annotations_test.go:187: Testing pass
--- PASS: TestAffinitySession (0.00s)
=== RUN   TestCors
    annotations_test.go:222: Testing pass true GET, PUT, POST, DELETE, PATCH, OPTIONS DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Authorization * true
    annotations_test.go:222: Testing pass true POST, GET, OPTIONS DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Authorization * false
    annotations_test.go:222: Testing pass true GET, PUT, POST, DELETE, PATCH, OPTIONS DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Authorization * false
    annotations_test.go:222: Testing pass false GET, PUT, POST, DELETE, PATCH, OPTIONS DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Authorization * true
    annotations_test.go:222: Testing pass false GET, PUT, POST, DELETE, PATCH, OPTIONS DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Authorization * true
--- PASS: TestCors (0.00s)
=== RUN   TestCustomHTTPErrors
--- PASS: TestCustomHTTPErrors (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations       (cached)
=== RUN   TestParse
--- PASS: TestParse (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/alias (cached)
=== RUN   TestIngressWithoutAuth
--- PASS: TestIngressWithoutAuth (0.00s)
=== RUN   TestIngressAuthBadAuthType
--- PASS: TestIngressAuthBadAuthType (0.00s)
=== RUN   TestInvalidIngressAuthNoSecret
--- PASS: TestInvalidIngressAuthNoSecret (0.00s)
=== RUN   TestIngressAuth
--- PASS: TestIngressAuth (0.00s)
=== RUN   TestIngressAuthWithoutSecret
--- PASS: TestIngressAuthWithoutSecret (0.00s)
=== RUN   TestIngressAuthInvalidSecretKey
--- PASS: TestIngressAuthInvalidSecretKey (0.00s)
=== RUN   TestDumpSecretAuthFile
--- PASS: TestDumpSecretAuthFile (0.00s)
=== RUN   TestDumpSecretAuthMap
--- PASS: TestDumpSecretAuthMap (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/auth  (cached)
=== RUN   TestAnnotations
--- PASS: TestAnnotations (0.00s)
=== RUN   TestHeaderAnnotations
--- PASS: TestHeaderAnnotations (0.00s)
=== RUN   TestCacheDurationAnnotations
--- PASS: TestCacheDurationAnnotations (0.00s)
=== RUN   TestParseStringToCacheDurations
--- PASS: TestParseStringToCacheDurations (0.00s)
=== RUN   TestProxySetHeaders
--- PASS: TestProxySetHeaders (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/authreq       (cached)
=== RUN   TestAnnotation
--- PASS: TestAnnotation (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/authreqglobal (cached)
=== RUN   TestAnnotations
--- PASS: TestAnnotations (0.00s)
=== RUN   TestInvalidAnnotations
--- PASS: TestInvalidAnnotations (0.00s)
=== RUN   TestEquals
--- PASS: TestEquals (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/authtls       (cached)
=== RUN   TestParseInvalidAnnotations
W0704 13:13:28.863164   17863 main.go:60] Protocol INVALID is not a valid value for the backend-protocol annotation. Using HTTP as protocol
--- PASS: TestParseInvalidAnnotations (0.00s)
=== RUN   TestParseAnnotations
--- PASS: TestParseAnnotations (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/backendprotocol       (cached)
=== RUN   TestCanaryInvalid
--- PASS: TestCanaryInvalid (0.00s)
=== RUN   TestAnnotations
--- PASS: TestAnnotations (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/canary        (cached)
=== RUN   TestIsValidClass
--- PASS: TestIsValidClass (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/class (cached)
=== RUN   TestParse
--- PASS: TestParse (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/clientbodybuffersize  (cached)
=== RUN   TestParse
--- PASS: TestParse (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/connection    (cached)
=== RUN   TestIngressCorsConfigValid
--- PASS: TestIngressCorsConfigValid (0.00s)
=== RUN   TestIngressCorsConfigInvalid
--- PASS: TestIngressCorsConfigInvalid (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/cors  (cached)
=== RUN   TestParseInvalidAnnotations
--- PASS: TestParseInvalidAnnotations (0.00s)
=== RUN   TestParseAnnotations
--- PASS: TestParseAnnotations (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/customhttperrors      (cached)
=== RUN   TestAnnotations
--- PASS: TestAnnotations (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/defaultbackend        (cached)
=== RUN   TestParseEmptyFastCGIAnnotations
--- PASS: TestParseEmptyFastCGIAnnotations (0.00s)
=== RUN   TestParseFastCGIIndexAnnotation
--- PASS: TestParseFastCGIIndexAnnotation (0.00s)
=== RUN   TestParseEmptyFastCGIParamsConfigMapAnnotation
--- PASS: TestParseEmptyFastCGIParamsConfigMapAnnotation (0.00s)
=== RUN   TestParseFastCGIInvalidParamsConfigMapAnnotation
--- PASS: TestParseFastCGIInvalidParamsConfigMapAnnotation (0.00s)
=== RUN   TestParseFastCGIParamsConfigMapAnnotationWithoutNS
--- PASS: TestParseFastCGIParamsConfigMapAnnotationWithoutNS (0.00s)
=== RUN   TestParseFastCGIParamsConfigMapAnnotationWithNS
--- PASS: TestParseFastCGIParamsConfigMapAnnotationWithNS (0.00s)
=== RUN   TestConfigEquality
--- PASS: TestConfigEquality (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/fastcgi       (cached)
=== RUN   TestGlobalRateLimiting
--- PASS: TestGlobalRateLimiting (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/globalratelimit       (cached)
=== RUN   TestParse
--- PASS: TestParse (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/http2pushpreload      (cached)
=== RUN   TestIngressInvalidInfluxDB
--- PASS: TestIngressInvalidInfluxDB (0.00s)
=== RUN   TestIngressInfluxDB
--- PASS: TestIngressInfluxDB (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/influxdb      (cached)
=== RUN   TestParseAnnotations
--- PASS: TestParseAnnotations (0.00s)
=== RUN   TestParseAnnotationsWithDefaultConfig
--- PASS: TestParseAnnotationsWithDefaultConfig (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/ipwhitelist   (cached)
=== RUN   TestParse
--- PASS: TestParse (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/loadbalancing (cached)
=== RUN   TestIngressAccessLogConfig
--- PASS: TestIngressAccessLogConfig (0.00s)
=== RUN   TestIngressRewriteLogConfig
--- PASS: TestIngressRewriteLogConfig (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/log   (cached)
=== RUN   TestParse
--- PASS: TestParse (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/mirror        (cached)
=== RUN   TestParse
--- PASS: TestParse (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/modsecurity   (cached)
=== RUN   TestIngressAnnotationOpentracingSetTrue
--- PASS: TestIngressAnnotationOpentracingSetTrue (0.00s)
=== RUN   TestIngressAnnotationOpentracingSetFalse
--- PASS: TestIngressAnnotationOpentracingSetFalse (0.00s)
=== RUN   TestIngressAnnotationOpentracingUnset
--- PASS: TestIngressAnnotationOpentracingUnset (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/opentracing   (cached)
=== RUN   TestGetBoolAnnotation
--- PASS: TestGetBoolAnnotation (0.00s)
=== RUN   TestGetStringAnnotation
--- PASS: TestGetStringAnnotation (0.00s)
=== RUN   TestGetIntAnnotation
--- PASS: TestGetIntAnnotation (0.00s)
=== RUN   TestStringToURL
--- PASS: TestStringToURL (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/parser        (cached)
=== RUN   TestPortInRedirect
--- PASS: TestPortInRedirect (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/portinredirect        (cached)
=== RUN   TestProxy
--- PASS: TestProxy (0.00s)
=== RUN   TestProxyWithNoAnnotation
--- PASS: TestProxyWithNoAnnotation (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/proxy (cached)
=== RUN   TestAnnotations
--- PASS: TestAnnotations (0.00s)
=== RUN   TestInvalidAnnotations
--- PASS: TestInvalidAnnotations (0.00s)
=== RUN   TestEquals
--- PASS: TestEquals (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/proxyssl      (cached)
=== RUN   TestWithoutAnnotations
--- PASS: TestWithoutAnnotations (0.00s)
=== RUN   TestRateLimiting
--- PASS: TestRateLimiting (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/ratelimit     (cached)
=== RUN   TestPermanentRedirectWithDefaultCode
--- PASS: TestPermanentRedirectWithDefaultCode (0.00s)
=== RUN   TestPermanentRedirectWithCustomCode
=== RUN   TestPermanentRedirectWithCustomCode/invalid_code
=== RUN   TestPermanentRedirectWithCustomCode/valid_code
--- PASS: TestPermanentRedirectWithCustomCode (0.00s)
    --- PASS: TestPermanentRedirectWithCustomCode/invalid_code (0.00s)
    --- PASS: TestPermanentRedirectWithCustomCode/valid_code (0.00s)
=== RUN   TestTemporalRedirect
--- PASS: TestTemporalRedirect (0.00s)
=== RUN   TestIsValidURL
--- PASS: TestIsValidURL (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/redirect      (cached)
=== RUN   TestWithoutAnnotations
--- PASS: TestWithoutAnnotations (0.00s)
=== RUN   TestRedirect
--- PASS: TestRedirect (0.00s)
=== RUN   TestSSLRedirect
--- PASS: TestSSLRedirect (0.00s)
=== RUN   TestForceSSLRedirect
--- PASS: TestForceSSLRedirect (0.00s)
=== RUN   TestAppRoot
=== RUN   TestAppRoot/Empty_path_should_return_an_error
=== RUN   TestAppRoot/Relative_paths_are_not_allowed
W0704 13:13:31.176773   20199 main.go:116] Annotation app-root contains an invalid value: parse "demo": invalid URI for request
=== RUN   TestAppRoot/Path_/_should_pass
=== RUN   TestAppRoot/Path_/demo_should_pass
--- PASS: TestAppRoot (0.00s)
    --- PASS: TestAppRoot/Empty_path_should_return_an_error (0.00s)
    --- PASS: TestAppRoot/Relative_paths_are_not_allowed (0.00s)
    --- PASS: TestAppRoot/Path_/_should_pass (0.00s)
    --- PASS: TestAppRoot/Path_/demo_should_pass (0.00s)
=== RUN   TestUseRegex
--- PASS: TestUseRegex (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/rewrite       (cached)
=== RUN   TestSatisfyParser
--- PASS: TestSatisfyParser (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/satisfy       (cached)
=== RUN   TestNoCA
--- PASS: TestNoCA (0.00s)
=== RUN   TestAnnotations
W0704 13:13:31.433726   20288 main.go:45] NOTE! secure-verify-ca-secret is not supported anymore. Please use proxy-ssl-secret instead
--- PASS: TestAnnotations (0.00s)
=== RUN   TestSecretNotFound
W0704 13:13:31.433786   20288 main.go:45] NOTE! secure-verify-ca-secret is not supported anymore. Please use proxy-ssl-secret instead
--- PASS: TestSecretNotFound (0.00s)
=== RUN   TestSecretOnNonSecure
W0704 13:13:31.433801   20288 main.go:45] NOTE! secure-verify-ca-secret is not supported anymore. Please use proxy-ssl-secret instead
--- PASS: TestSecretOnNonSecure (0.00s)
=== RUN   TestUnsupportedAnnotation
W0704 13:13:31.433819   20288 main.go:45] NOTE! secure-verify-ca-secret is not supported anymore. Please use proxy-ssl-secret instead
--- PASS: TestUnsupportedAnnotation (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/secureupstream        (cached)
=== RUN   TestParse
--- PASS: TestParse (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/serversnippet (cached)
=== RUN   TestIngressAnnotationServiceUpstreamEnabled
--- PASS: TestIngressAnnotationServiceUpstreamEnabled (0.00s)
=== RUN   TestIngressAnnotationServiceUpstreamSetFalse
--- PASS: TestIngressAnnotationServiceUpstreamSetFalse (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/serviceupstream       (cached)
=== RUN   TestIngressAffinityCookieConfig
--- PASS: TestIngressAffinityCookieConfig (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/sessionaffinity       (cached)
=== RUN   TestParse
--- PASS: TestParse (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/snippet       (cached)
=== RUN   TestParse
--- PASS: TestParse (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/sslcipher     (cached)
=== RUN   TestParseAnnotations
--- PASS: TestParseAnnotations (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/sslpassthrough        (cached)
=== RUN   TestParse
--- PASS: TestParse (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/upstreamhashby        (cached)
=== RUN   TestParse
--- PASS: TestParse (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/upstreamvhost (cached)
=== RUN   TestParse
--- PASS: TestParse (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/annotations/xforwardedprefix      (cached)
=== RUN   TestNginxCheck
=== RUN   TestNginxCheck/health_path:_/healthz
=== RUN   TestNginxCheck/health_path:_/healthz/no_pid_or_process
=== RUN   TestNginxCheck/health_path:_/healthz/no_process
=== RUN   TestNginxCheck/health_path:_/healthz/valid_request
=== RUN   TestNginxCheck/health_path:_/healthz/bad_pid
=== RUN   TestNginxCheck/health_path:_/not-healthz
=== RUN   TestNginxCheck/health_path:_/not-healthz/no_pid_or_process
=== RUN   TestNginxCheck/health_path:_/not-healthz/no_process
=== RUN   TestNginxCheck/health_path:_/not-healthz/valid_request
=== RUN   TestNginxCheck/health_path:_/not-healthz/bad_pid
--- PASS: TestNginxCheck (3.50s)
    --- PASS: TestNginxCheck/health_path:_/healthz (3.50s)
        --- PASS: TestNginxCheck/health_path:_/healthz/no_pid_or_process (0.00s)
        --- PASS: TestNginxCheck/health_path:_/healthz/no_process (0.00s)
        --- PASS: TestNginxCheck/health_path:_/healthz/valid_request (0.00s)
        --- PASS: TestNginxCheck/health_path:_/healthz/bad_pid (0.00s)
    --- PASS: TestNginxCheck/health_path:_/not-healthz (0.00s)
        --- PASS: TestNginxCheck/health_path:_/not-healthz/no_pid_or_process (0.00s)
        --- PASS: TestNginxCheck/health_path:_/not-healthz/no_process (0.00s)
        --- PASS: TestNginxCheck/health_path:_/not-healthz/valid_request (0.00s)
        --- PASS: TestNginxCheck/health_path:_/not-healthz/bad_pid (0.00s)
=== RUN   TestCheckIngress
=== RUN   TestCheckIngress/When_the_ingress_class_differs_from_nginx
W0704 13:13:36.968688   20923 controller.go:223] ignoring ingress test-ingress in user-namespace based on annotation kubernetes.io/ingress.class
=== RUN   TestCheckIngress/when_the_class_is_the_nginx_one
=== RUN   TestCheckIngress/when_the_class_is_the_nginx_one/When_the_hostname_is_updated
=== RUN   TestCheckIngress/when_the_class_is_the_nginx_one/When_nginx_test_returns_an_error
=== RUN   TestCheckIngress/when_the_class_is_the_nginx_one/When_the_default_annotation_prefix_is_used_despite_an_override
=== RUN   TestCheckIngress/when_the_class_is_the_nginx_one/When_a_new_catch-all_ingress_is_being_created_despite_catch-alls_being_disabled_
=== RUN   TestCheckIngress/when_the_class_is_the_nginx_one/When_the_ingress_is_in_a_different_namespace_than_the_watched_one
W0704 13:13:36.969823   20923 controller.go:228] ignoring ingress test-ingress in namespace test-namespace different from the namespace watched other-namespace
=== RUN   TestCheckIngress/When_the_ingress_is_marked_as_deleted
--- PASS: TestCheckIngress (0.12s)
    --- PASS: TestCheckIngress/When_the_ingress_class_differs_from_nginx (0.00s)
    --- PASS: TestCheckIngress/when_the_class_is_the_nginx_one (0.00s)
        --- PASS: TestCheckIngress/when_the_class_is_the_nginx_one/When_the_hostname_is_updated (0.00s)
        --- PASS: TestCheckIngress/when_the_class_is_the_nginx_one/When_nginx_test_returns_an_error (0.00s)
        --- PASS: TestCheckIngress/when_the_class_is_the_nginx_one/When_the_default_annotation_prefix_is_used_despite_an_override (0.00s)
        --- PASS: TestCheckIngress/when_the_class_is_the_nginx_one/When_a_new_catch-all_ingress_is_being_created_despite_catch-alls_being_disabled_ (0.00s)
        --- PASS: TestCheckIngress/when_the_class_is_the_nginx_one/When_the_ingress_is_in_a_different_namespace_than_the_watched_one (0.00s)
    --- PASS: TestCheckIngress/When_the_ingress_is_marked_as_deleted (0.00s)
=== RUN   TestMergeAlternativeBackends
=== RUN   TestMergeAlternativeBackends/catch-all_alternative_backend_has_no_server_and_embeds_into_matching_real_backend
=== RUN   TestMergeAlternativeBackends/catch-all_alternative_backend_does_not_merge_into_itself
W0704 13:13:36.971557   20923 controller.go:1375] unable to find real backend for alternative backend example-http-svc-canary-80. Deleting.
=== RUN   TestMergeAlternativeBackends/non-host_canary_ingress_use_default_server_name_as_host_to_merge
=== RUN   TestMergeAlternativeBackends/alternative_backend_has_no_server_and_embeds_into_matching_real_backend
=== RUN   TestMergeAlternativeBackends/alternative_backend_merges_with_the_correct_real_backend_when_multiple_are_present
=== RUN   TestMergeAlternativeBackends/alternative_backend_does_not_merge_into_itself
E0704 13:13:36.971627   20923 controller.go:1402] cannot merge alternative backend example-http-svc-canary-80 into hostname example.com that does not exist
--- PASS: TestMergeAlternativeBackends (0.00s)
    --- PASS: TestMergeAlternativeBackends/catch-all_alternative_backend_has_no_server_and_embeds_into_matching_real_backend (0.00s)
    --- PASS: TestMergeAlternativeBackends/catch-all_alternative_backend_does_not_merge_into_itself (0.00s)
    --- PASS: TestMergeAlternativeBackends/non-host_canary_ingress_use_default_server_name_as_host_to_merge (0.00s)
    --- PASS: TestMergeAlternativeBackends/alternative_backend_has_no_server_and_embeds_into_matching_real_backend (0.00s)
    --- PASS: TestMergeAlternativeBackends/alternative_backend_merges_with_the_correct_real_backend_when_multiple_are_present (0.00s)
    --- PASS: TestMergeAlternativeBackends/alternative_backend_does_not_merge_into_itself (0.00s)
=== RUN   TestExtractTLSSecretName
=== RUN   TestExtractTLSSecretName/ingress_tls,_no_host,_wildcard_cert_with_matching_cn
=== RUN   TestExtractTLSSecretName/ingress_tls,_hosts,_matching_cert_cn
=== RUN   TestExtractTLSSecretName/ingress_tls,_hosts,_matching_cert_cn,_uppercase_host
=== RUN   TestExtractTLSSecretName/nil_ingress
=== RUN   TestExtractTLSSecretName/empty_ingress
=== RUN   TestExtractTLSSecretName/ingress_tls,_nil_secret
=== RUN   TestExtractTLSSecretName/ingress_tls,_no_host,_matching_cert_cn
--- PASS: TestExtractTLSSecretName (0.00s)
    --- PASS: TestExtractTLSSecretName/ingress_tls,_no_host,_wildcard_cert_with_matching_cn (0.00s)
    --- PASS: TestExtractTLSSecretName/ingress_tls,_hosts,_matching_cert_cn (0.00s)
    --- PASS: TestExtractTLSSecretName/ingress_tls,_hosts,_matching_cert_cn,_uppercase_host (0.00s)
    --- PASS: TestExtractTLSSecretName/nil_ingress (0.00s)
    --- PASS: TestExtractTLSSecretName/empty_ingress (0.00s)
    --- PASS: TestExtractTLSSecretName/ingress_tls,_nil_secret (0.00s)
    --- PASS: TestExtractTLSSecretName/ingress_tls,_no_host,_matching_cert_cn (0.00s)
=== RUN   TestGetBackendServers
W0704 13:13:37.015521   20923 controller.go:836] Error creating upstream "example-http-svc-canary-80": no object matching key "example/http-svc-canary" in local store
W0704 13:13:37.015554   20923 controller.go:842] Error obtaining Service "example/http-svc-canary": no object matching key "example/http-svc-canary" in local store
W0704 13:13:37.105586   20923 controller.go:836] Error creating upstream "example-http-svc-canary-80": no object matching key "example/http-svc-canary" in local store
W0704 13:13:37.105609   20923 controller.go:842] Error obtaining Service "example/http-svc-canary": no object matching key "example/http-svc-canary" in local store
W0704 13:13:37.105616   20923 controller.go:836] Error creating upstream "example-http-svc-80": no object matching key "example/http-svc" in local store
W0704 13:13:37.105619   20923 controller.go:842] Error obtaining Service "example/http-svc": no object matching key "example/http-svc" in local store
W0704 13:13:37.231625   20923 controller.go:900] Error obtaining Endpoints for Service "example/http-svc-canary": no object matching key "example/http-svc-canary" in local store
W0704 13:13:37.271364   20923 controller.go:900] Error obtaining Endpoints for Service "example/http-svc": no object matching key "example/http-svc" in local store
W0704 13:13:37.271397   20923 controller.go:900] Error obtaining Endpoints for Service "example/http-svc-canary": no object matching key "example/http-svc-canary" in local store
W0704 13:13:37.465379   20923 controller.go:900] Error obtaining Endpoints for Service "example/http-svc-1": no object matching key "example/http-svc-1" in local store
W0704 13:13:37.465406   20923 controller.go:900] Error obtaining Endpoints for Service "example/http-svc-2": no object matching key "example/http-svc-2" in local store
W0704 13:13:37.465434   20923 controller.go:1414] alternative upstream example-http-svc-1-80 in Ingress example/example-b-canary is primary upstream in Other Ingress for location example.com/a!
W0704 13:13:37.613685   20923 controller.go:900] Error obtaining Endpoints for Service "proxyssl/path1-svc": no object matching key "proxyssl/path1-svc" in local store
W0704 13:13:37.613708   20923 controller.go:900] Error obtaining Endpoints for Service "proxyssl/path2-svc": no object matching key "proxyssl/path2-svc" in local store
W0704 13:13:37.704441   20923 controller.go:900] Error obtaining Endpoints for Service "proxyssl/path1-svc": no object matching key "proxyssl/path1-svc" in local store
W0704 13:13:37.704466   20923 controller.go:900] Error obtaining Endpoints for Service "proxyssl/path2-svc": no object matching key "proxyssl/path2-svc" in local store
--- PASS: TestGetBackendServers (0.73s)
=== RUN   TestGetEndpoints
=== RUN   TestGetEndpoints/no_service_should_return_0_endpoint
=== RUN   TestGetEndpoints/no_service_port_should_return_0_endpoint
=== RUN   TestGetEndpoints/a_service_without_endpoint_should_return_0_endpoint
=== RUN   TestGetEndpoints/a_service_type_ServiceTypeExternalName_service_with_an_invalid_port_should_return_0_endpoint
E0704 13:13:37.704647   20923 endpoints.go:65] Invalid DNS name : [a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')]
=== RUN   TestGetEndpoints/a_service_type_ServiceTypeExternalName_service_with_localhost_in_name_should_return_0_endpoint
E0704 13:13:37.704684   20923 endpoints.go:55] Invalid attempt to use localhost name localhost in ""
=== RUN   TestGetEndpoints/a_service_type_ServiceTypeExternalName_service_with_127.0.0.1_in_name_should_return_0_endpoint
E0704 13:13:37.704718   20923 endpoints.go:55] Invalid attempt to use localhost name 127.0.0.1 in ""
=== RUN   TestGetEndpoints/a_service_type_ServiceTypeExternalName_with_a_valid_port_should_return_one_endpoint
=== RUN   TestGetEndpoints/a_service_type_ServiceTypeExternalName_with_an_trailing_dot_ExternalName_value_should_return_one_endpoints
=== RUN   TestGetEndpoints/a_service_type_ServiceTypeExternalName_with_an_invalid_ExternalName_value_should_no_return_endpoints
E0704 13:13:37.704818   20923 endpoints.go:65] Invalid DNS name 1#invalid.hostname: [a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')]
=== RUN   TestGetEndpoints/should_return_no_endpoint_when_there_is_an_error_searching_for_endpoints
W0704 13:13:37.704871   20923 endpoints.go:79] Error obtaining Endpoints for Service "": unexpected error
=== RUN   TestGetEndpoints/should_return_no_endpoint_when_the_protocol_does_not_match
=== RUN   TestGetEndpoints/should_return_no_endpoint_when_there_is_no_ready_Addresses
=== RUN   TestGetEndpoints/should_return_no_endpoint_when_the_name_of_the_port_name_do_not_match_any_port_in_the_endpoint_Subsets
=== RUN   TestGetEndpoints/should_return_one_endpoint_when_the_name_of_the_port_name_match_a_port_in_the_endpoint_Subsets
=== RUN   TestGetEndpoints/should_return_one_endpoint_when_the_name_of_the_port_name_match_more_than_one_port_in_the_endpoint_Subsets
--- PASS: TestGetEndpoints (0.00s)
    --- PASS: TestGetEndpoints/no_service_should_return_0_endpoint (0.00s)
    --- PASS: TestGetEndpoints/no_service_port_should_return_0_endpoint (0.00s)
    --- PASS: TestGetEndpoints/a_service_without_endpoint_should_return_0_endpoint (0.00s)
    --- PASS: TestGetEndpoints/a_service_type_ServiceTypeExternalName_service_with_an_invalid_port_should_return_0_endpoint (0.00s)
    --- PASS: TestGetEndpoints/a_service_type_ServiceTypeExternalName_service_with_localhost_in_name_should_return_0_endpoint (0.00s)
    --- PASS: TestGetEndpoints/a_service_type_ServiceTypeExternalName_service_with_127.0.0.1_in_name_should_return_0_endpoint (0.00s)
    --- PASS: TestGetEndpoints/a_service_type_ServiceTypeExternalName_with_a_valid_port_should_return_one_endpoint (0.00s)
    --- PASS: TestGetEndpoints/a_service_type_ServiceTypeExternalName_with_an_trailing_dot_ExternalName_value_should_return_one_endpoints (0.00s)
    --- PASS: TestGetEndpoints/a_service_type_ServiceTypeExternalName_with_an_invalid_ExternalName_value_should_no_return_endpoints (0.00s)
    --- PASS: TestGetEndpoints/should_return_no_endpoint_when_there_is_an_error_searching_for_endpoints (0.00s)
    --- PASS: TestGetEndpoints/should_return_no_endpoint_when_the_protocol_does_not_match (0.00s)
    --- PASS: TestGetEndpoints/should_return_no_endpoint_when_there_is_no_ready_Addresses (0.00s)
    --- PASS: TestGetEndpoints/should_return_no_endpoint_when_the_name_of_the_port_name_do_not_match_any_port_in_the_endpoint_Subsets (0.00s)
    --- PASS: TestGetEndpoints/should_return_one_endpoint_when_the_name_of_the_port_name_match_a_port_in_the_endpoint_Subsets (0.00s)
    --- PASS: TestGetEndpoints/should_return_one_endpoint_when_the_name_of_the_port_name_match_more_than_one_port_in_the_endpoint_Subsets (0.00s)
=== RUN   TestIsDynamicConfigurationEnough
--- PASS: TestIsDynamicConfigurationEnough (0.00s)
=== RUN   TestConfigureDynamically
--- PASS: TestConfigureDynamically (0.00s)
=== RUN   TestConfigureCertificates
--- PASS: TestConfigureCertificates (0.00s)
=== RUN   TestNginxHashBucketSize
--- PASS: TestNginxHashBucketSize (0.00s)
=== RUN   TestNextPowerOf2
--- PASS: TestNextPowerOf2 (0.00s)
=== RUN   TestCleanTempNginxCfg
--- PASS: TestCleanTempNginxCfg (0.00s)
=== RUN   TestRlimitMaxNumFiles
--- PASS: TestRlimitMaxNumFiles (0.00s)
=== RUN   TestSysctlSomaxconn
--- PASS: TestSysctlSomaxconn (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/controller        (cached)
?       k8s.io/ingress-nginx/internal/ingress/controller/config [no test files]
=== RUN   TestIsRespawnIfRequired
W0704 13:13:20.840882    7344 nginx.go:34]
-------------------------------------------------------------------------------
NGINX master process died (0): exit status 0
-------------------------------------------------------------------------------
--- PASS: TestIsRespawnIfRequired (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/controller/process        (cached)
=== RUN   TestSSLCertTracker
--- PASS: TestSSLCertTracker (0.00s)
=== RUN   TestObjectRefMapOperations
--- PASS: TestObjectRefMapOperations (0.00s)
=== RUN   TestStore
=== RUN   TestStore/should_return_an_error_searching_for_non_existing_objects
I0704 14:30:11.894237    1905 event.go:282] Event(v1.ObjectReference{Kind:"ConfigMap", Namespace:"store-test-1625409011", Name:"config", UID:"e1fe294a-fa98-4e89-9b5b-5f6a41715224", APIVersion:"v1", ResourceVersion:"47", FieldPath:""}): type: 'Normal' reason: 'CREATE' ConfigMap store-test-1625409011/config
=== RUN   TestStore/should_return_one_event_for_add,_update_and_delete_of_ingress
I0704 14:30:13.101138    1905 event.go:282] Event(v1.ObjectReference{Kind:"ConfigMap", Namespace:"store-test-1625409013", Name:"config", UID:"8d50fd79-4db8-4746-9010-d351ae2b0b75", APIVersion:"v1", ResourceVersion:"51", FieldPath:""}): type: 'Normal' reason: 'CREATE' ConfigMap store-test-1625409013/config
    store_test.go:170: Ingress &Ingress{ObjectMeta:{dummy  store-test-1625409013    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},Spec:IngressSpec{DefaultBackend:nil,TLS:[]IngressTLS{},Rules:[]IngressRule{IngressRule{Host:dummy,IngressRuleValue:IngressRuleValue{HTTP:&HTTPIngressRuleValue{Paths:[]HTTPIngressPath{HTTPIngressPath{Path:/,Backend:IngressBackend{Resource:nil,Service:&IngressServiceBackend{Name:http-svc,Port:ServiceBackendPort{Name:,Number:80,},},},PathType:*Prefix,},},},},},},IngressClassName:nil,},Status:IngressStatus{LoadBalancer:{[]},},} not found, creating
    store_test.go:170: Ingress &Ingress{ObjectMeta:{dummy  store-test-1625409013    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},Spec:IngressSpec{DefaultBackend:nil,TLS:[]IngressTLS{},Rules:[]IngressRule{IngressRule{Host:dummy,IngressRuleValue:IngressRuleValue{HTTP:&HTTPIngressRuleValue{Paths:[]HTTPIngressPath{HTTPIngressPath{Path:/,Backend:IngressBackend{Resource:nil,Service:&IngressServiceBackend{Name:http-svc,Port:ServiceBackendPort{Name:,Number:80,},},},PathType:*Prefix,},},},},},},IngressClassName:nil,},Status:IngressStatus{LoadBalancer:{[]},},} created
I0704 14:30:14.304893    1905 event.go:282] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"store-test-1625409013", Name:"dummy", UID:"ce42a4a2-ebbc-4a24-ab83-4d119406ce4e", APIVersion:"networking.k8s.io/v1", ResourceVersion:"53", FieldPath:""}): type: 'Normal' reason: 'Sync' Scheduled for sync
    store_test.go:209: Ingress &Ingress{ObjectMeta:{custom-class  store-test-1625409013    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[kubernetes.io/ingress.class:something] [] []  []},Spec:IngressSpec{DefaultBackend:nil,TLS:[]IngressTLS{},Rules:[]IngressRule{IngressRule{Host:dummy,IngressRuleValue:IngressRuleValue{HTTP:&HTTPIngressRuleValue{Paths:[]HTTPIngressPath{HTTPIngressPath{Path:/,Backend:IngressBackend{Resource:nil,Service:&IngressServiceBackend{Name:http-svc,Port:ServiceBackendPort{Name:,Number:80,},},},PathType:*Prefix,},},},},},},IngressClassName:nil,},Status:IngressStatus{LoadBalancer:{[]},},} not found, creating
I0704 14:30:17.309862    1905 store.go:352] "Ignoring ingress" ingress="store-test-1625409013/custom-class" kubernetes.io/ingress.class="" ingressClassName=""
    store_test.go:209: Ingress &Ingress{ObjectMeta:{custom-class  store-test-1625409013    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[kubernetes.io/ingress.class:something] [] []  []},Spec:IngressSpec{DefaultBackend:nil,TLS:[]IngressTLS{},Rules:[]IngressRule{IngressRule{Host:dummy,IngressRuleValue:IngressRuleValue{HTTP:&HTTPIngressRuleValue{Paths:[]HTTPIngressPath{HTTPIngressPath{Path:/,Backend:IngressBackend{Resource:nil,Service:&IngressServiceBackend{Name:http-svc,Port:ServiceBackendPort{Name:,Number:80,},},},PathType:*Prefix,},},},},},},IngressClassName:nil,},Status:IngressStatus{LoadBalancer:{[]},},} created
I0704 14:30:17.311154    1905 event.go:282] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"store-test-1625409013", Name:"dummy", UID:"ce42a4a2-ebbc-4a24-ab83-4d119406ce4e", APIVersion:"networking.k8s.io/v1", ResourceVersion:"56", FieldPath:""}): type: 'Normal' reason: 'Sync' Scheduled for sync
    store_test.go:274: Ingress &Ingress{ObjectMeta:{custom-class  store-test-1625409013 /apis/networking.k8s.io/v1/namespaces/store-test-1625409013/ingresses/custom-class f89048b3-a54a-4695-8a5d-8beeae843d47 55 1 2021-07-04 14:30:17 +0000 UTC <nil> <nil> map[] map[kubernetes.io/ingress.class:something] [] []  [{store.test Update networking.k8s.io/v1 2021-07-04 14:30:17 +0000 UTC FieldsV1 {"f:metadata":{"f:annotations":{".":{},"f:kubernetes.io/ingress.class":{}}},"f:spec":{"f:rules":{}}}}]},Spec:IngressSpec{DefaultBackend:nil,TLS:[]IngressTLS{},Rules:[]IngressRule{IngressRule{Host:dummy,IngressRuleValue:IngressRuleValue{HTTP:&HTTPIngressRuleValue{Paths:[]HTTPIngressPath{HTTPIngressPath{Path:/,Backend:IngressBackend{Resource:nil,Service:&IngressServiceBackend{Name:http-svc,Port:ServiceBackendPort{Name:,Number:80,},},},PathType:*Prefix,},},},},},},IngressClassName:nil,},Status:IngressStatus{LoadBalancer:{[]},},} deleted
=== RUN   TestStore/should_not_receive_updates_for_ingress_with_invalid_class
I0704 14:30:23.325037    1905 event.go:282] Event(v1.ObjectReference{Kind:"ConfigMap", Namespace:"store-test-1625409023", Name:"config", UID:"c4911a0a-1c37-4912-b0fd-db67bdd207e6", APIVersion:"v1", ResourceVersion:"63", FieldPath:""}): type: 'Normal' reason: 'CREATE' ConfigMap store-test-1625409023/config
    store_test.go:328: Ingress &Ingress{ObjectMeta:{custom-class  store-test-1625409023    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[kubernetes.io/ingress.class:something] [] []  []},Spec:IngressSpec{DefaultBackend:nil,TLS:[]IngressTLS{},Rules:[]IngressRule{IngressRule{Host:dummy,IngressRuleValue:IngressRuleValue{HTTP:&HTTPIngressRuleValue{Paths:[]HTTPIngressPath{HTTPIngressPath{Path:/,Backend:IngressBackend{Resource:nil,Service:&IngressServiceBackend{Name:http-svc,Port:ServiceBackendPort{Name:,Number:80,},},},PathType:*Prefix,},},},},},},IngressClassName:nil,},Status:IngressStatus{LoadBalancer:{[]},},} not found, creating
    store_test.go:328: Ingress &Ingress{ObjectMeta:{custom-class  store-test-1625409023    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[kubernetes.io/ingress.class:something] [] []  []},Spec:IngressSpec{DefaultBackend:nil,TLS:[]IngressTLS{},Rules:[]IngressRule{IngressRule{Host:dummy,IngressRuleValue:IngressRuleValue{HTTP:&HTTPIngressRuleValue{Paths:[]HTTPIngressPath{HTTPIngressPath{Path:/,Backend:IngressBackend{Resource:nil,Service:&IngressServiceBackend{Name:http-svc,Port:ServiceBackendPort{Name:,Number:80,},},},PathType:*Prefix,},},},},},},IngressClassName:nil,},Status:IngressStatus{LoadBalancer:{[]},},} created
I0704 14:30:24.527241    1905 store.go:352] "Ignoring ingress" ingress="store-test-1625409023/custom-class" kubernetes.io/ingress.class="" ingressClassName=""
=== RUN   TestStore/should_not_receive_events_from_secret_not_referenced_from_ingress
I0704 14:30:30.540063    1905 event.go:282] Event(v1.ObjectReference{Kind:"ConfigMap", Namespace:"store-test-1625409030", Name:"config", UID:"e430c452-9b77-4239-bed6-c577546832da", APIVersion:"v1", ResourceVersion:"69", FieldPath:""}): type: 'Normal' reason: 'CREATE' ConfigMap store-test-1625409030/config
=== RUN   TestStore/should_receive_events_from_secret_referenced_from_ingress
I0704 14:30:34.793288    1905 event.go:282] Event(v1.ObjectReference{Kind:"ConfigMap", Namespace:"store-test-1625409034", Name:"config", UID:"c1aae06b-b579-4a72-b9ed-26da86df2e6b", APIVersion:"v1", ResourceVersion:"76", FieldPath:""}): type: 'Normal' reason: 'CREATE' ConfigMap store-test-1625409034/config
    store_test.go:524: Ingress &Ingress{ObjectMeta:{ingress-with-secret  store-test-1625409034    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},Spec:IngressSpec{DefaultBackend:&IngressBackend{Resource:nil,Service:&IngressServiceBackend{Name:http-svc,Port:ServiceBackendPort{Name:,Number:80,},},},TLS:[]IngressTLS{IngressTLS{Hosts:[],SecretName:referenced,},},Rules:[]IngressRule{},IngressClassName:nil,},Status:IngressStatus{LoadBalancer:{[]},},} not found, creating
    store_test.go:524: Ingress &Ingress{ObjectMeta:{ingress-with-secret  store-test-1625409034    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},Spec:IngressSpec{DefaultBackend:&IngressBackend{Resource:nil,Service:&IngressServiceBackend{Name:http-svc,Port:ServiceBackendPort{Name:,Number:80,},},},TLS:[]IngressTLS{IngressTLS{Hosts:[],SecretName:referenced,},},Rules:[]IngressRule{},IngressClassName:nil,},Status:IngressStatus{LoadBalancer:{[]},},} created
I0704 14:30:35.995698    1905 event.go:282] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"store-test-1625409034", Name:"ingress-with-secret", UID:"8b38986d-057b-4b7b-9156-010d79a2a00b", APIVersion:"networking.k8s.io/v1", ResourceVersion:"78", FieldPath:""}): type: 'Normal' reason: 'Sync' Scheduled for sync
W0704 14:30:35.995729    1905 backend_ssl.go:46] Error obtaining X.509 certificate: no object matching key "store-test-1625409034/referenced" in local store
I0704 14:30:38.042497    1905 store.go:426] "Secret was added and it is used in ingress annotations. Parsing" secret="store-test-1625409034/referenced"
I0704 14:30:38.042773    1905 backend_ssl.go:66] "Adding secret to local store" name="store-test-1625409034/referenced"
I0704 14:30:43.046515    1905 store.go:491] "secret was deleted and it is used in ingress annotations. Parsing" secret="store-test-1625409034/referenced"
    store_test.go:584: Ingress &Ingress{ObjectMeta:{ingress-with-secret  store-test-1625409034 /apis/networking.k8s.io/v1/namespaces/store-test-1625409034/ingresses/ingress-with-secret 8b38986d-057b-4b7b-9156-010d79a2a00b 78 1 2021-07-04 14:30:35 +0000 UTC <nil> <nil> map[] map[] [] []  [{store.test Update networking.k8s.io/v1 2021-07-04 14:30:35 +0000 UTC FieldsV1 {"f:spec":{"f:defaultBackend":{".":{},"f:service":{".":{},"f:name":{},"f:port":{".":{},"f:number":{}}}},"f:tls":{}}}}]},Spec:IngressSpec{DefaultBackend:&IngressBackend{Resource:nil,Service:&IngressServiceBackend{Name:http-svc,Port:ServiceBackendPort{Name:,Number:80,},},},TLS:[]IngressTLS{IngressTLS{Hosts:[],SecretName:referenced,},},Rules:[]IngressRule{},IngressClassName:nil,},Status:IngressStatus{LoadBalancer:{[]},},} deleted
=== RUN   TestStore/should_create_an_ingress_with_a_secret_which_does_not_exist
I0704 14:30:44.055302    1905 event.go:282] Event(v1.ObjectReference{Kind:"ConfigMap", Namespace:"store-test-1625409044", Name:"config", UID:"23359e28-1faa-449f-9779-b91f07eab94c", APIVersion:"v1", ResourceVersion:"86", FieldPath:""}): type: 'Normal' reason: 'CREATE' ConfigMap store-test-1625409044/config
    store_test.go:636: Ingress &Ingress{ObjectMeta:{ingress-with-secret  store-test-1625409044    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},Spec:IngressSpec{DefaultBackend:nil,TLS:[]IngressTLS{IngressTLS{Hosts:[ingress-with-secret],SecretName:ingress-with-secret,},},Rules:[]IngressRule{IngressRule{Host:ingress-with-secret,IngressRuleValue:IngressRuleValue{HTTP:&HTTPIngressRuleValue{Paths:[]HTTPIngressPath{HTTPIngressPath{Path:/,Backend:IngressBackend{Resource:nil,Service:&IngressServiceBackend{Name:http-svc,Port:ServiceBackendPort{Name:,Number:80,},},},PathType:*Prefix,},},},},},},IngressClassName:nil,},Status:IngressStatus{LoadBalancer:{[]},},} not found, creating
    store_test.go:636: Ingress &Ingress{ObjectMeta:{ingress-with-secret  store-test-1625409044    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},Spec:IngressSpec{DefaultBackend:nil,TLS:[]IngressTLS{IngressTLS{Hosts:[ingress-with-secret],SecretName:ingress-with-secret,},},Rules:[]IngressRule{IngressRule{Host:ingress-with-secret,IngressRuleValue:IngressRuleValue{HTTP:&HTTPIngressRuleValue{Paths:[]HTTPIngressPath{HTTPIngressPath{Path:/,Backend:IngressBackend{Resource:nil,Service:&IngressServiceBackend{Name:http-svc,Port:ServiceBackendPort{Name:,Number:80,},},},PathType:*Prefix,},},},},},},IngressClassName:nil,},Status:IngressStatus{LoadBalancer:{[]},},} created
W0704 14:30:45.257682    1905 backend_ssl.go:46] Error obtaining X.509 certificate: no object matching key "store-test-1625409044/ingress-with-secret" in local store
I0704 14:30:45.257686    1905 event.go:282] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"store-test-1625409044", Name:"ingress-with-secret", UID:"9cbdf3ac-8c2a-46ef-bccb-dea4e2ac949f", APIVersion:"networking.k8s.io/v1", ResourceVersion:"88", FieldPath:""}): type: 'Normal' reason: 'Sync' Scheduled for sync
I0704 14:30:50.305020    1905 store.go:426] "Secret was added and it is used in ingress annotations. Parsing" secret="store-test-1625409044/ingress-with-secret"
I0704 14:30:50.305292    1905 backend_ssl.go:66] "Adding secret to local store" name="store-test-1625409044/ingress-with-secret"
    store_test.go:700: Ingress &Ingress{ObjectMeta:{ingress-with-secret  store-test-1625409044 /apis/networking.k8s.io/v1/namespaces/store-test-1625409044/ingresses/ingress-with-secret 9cbdf3ac-8c2a-46ef-bccb-dea4e2ac949f 88 1 2021-07-04 14:30:45 +0000 UTC <nil> <nil> map[] map[] [] []  [{store.test Update networking.k8s.io/v1 2021-07-04 14:30:45 +0000 UTC FieldsV1 {"f:spec":{"f:rules":{},"f:tls":{}}}}]},Spec:IngressSpec{DefaultBackend:nil,TLS:[]IngressTLS{IngressTLS{Hosts:[ingress-with-secret],SecretName:ingress-with-secret,},},Rules:[]IngressRule{IngressRule{Host:ingress-with-secret,IngressRuleValue:IngressRuleValue{HTTP:&HTTPIngressRuleValue{Paths:[]HTTPIngressPath{HTTPIngressPath{Path:/,Backend:IngressBackend{Resource:nil,Service:&IngressServiceBackend{Name:http-svc,Port:ServiceBackendPort{Name:,Number:80,},},},PathType:*Prefix,},},},},},},IngressClassName:nil,},Status:IngressStatus{LoadBalancer:{[]},},} deleted
--- PASS: TestStore (42.67s)
    --- PASS: TestStore/should_return_an_error_searching_for_non_existing_objects (1.21s)
    --- PASS: TestStore/should_return_one_event_for_add,_update_and_delete_of_ingress (10.22s)
    --- PASS: TestStore/should_not_receive_updates_for_ingress_with_invalid_class (7.21s)
    --- PASS: TestStore/should_not_receive_events_from_secret_not_referenced_from_ingress (4.25s)
    --- PASS: TestStore/should_receive_events_from_secret_referenced_from_ingress (9.26s)
    --- PASS: TestStore/should_create_an_ingress_with_a_secret_which_does_not_exist (6.26s)
=== RUN   TestUpdateSecretIngressMap
=== RUN   TestUpdateSecretIngressMap/with_TLS_secret
=== RUN   TestUpdateSecretIngressMap/with_annotation_in_simple_name_format
=== RUN   TestUpdateSecretIngressMap/with_annotation_in_namespace/name_format
=== RUN   TestUpdateSecretIngressMap/with_annotation_in_invalid_format
E0704 14:30:50.814902    1905 store.go:699] error reading secret reference in annotation "auth-secret": unexpected key format: "ns/name/garbage"
--- PASS: TestUpdateSecretIngressMap (0.00s)
    --- PASS: TestUpdateSecretIngressMap/with_TLS_secret (0.00s)
    --- PASS: TestUpdateSecretIngressMap/with_annotation_in_simple_name_format (0.00s)
    --- PASS: TestUpdateSecretIngressMap/with_annotation_in_namespace/name_format (0.00s)
    --- PASS: TestUpdateSecretIngressMap/with_annotation_in_invalid_format (0.00s)
=== RUN   TestListIngresses
--- PASS: TestListIngresses (0.00s)
=== RUN   TestWriteSSLSessionTicketKey
--- PASS: TestWriteSSLSessionTicketKey (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/controller/store  42.707s
=== RUN   TestFilterErrors
W0704 13:13:32.325044   20774 configmap.go:411] error code 200 is not valid for custom error pages
W0704 13:13:32.325080   20774 configmap.go:411] error code 999 is not valid for custom error pages
--- PASS: TestFilterErrors (0.00s)
=== RUN   TestProxyTimeoutParsing
W0704 13:13:32.329914   20774 configmap.go:321] proxy-protocol-header-timeout of 3zxs encountered an error while being parsed time: unknown unit "zxs" in duration "3zxs". Switching to use default value instead.
--- PASS: TestProxyTimeoutParsing (0.01s)
=== RUN   TestMergeConfigMapToStruct
W0704 13:13:32.331005   20774 configmap.go:150] demo is not a valid http code: strconv.Atoi: parsing "demo": invalid syntax
W0704 13:13:32.331044   20774 configmap.go:190] 3.3.3 is not a valid textual representation of an IP address
W0704 13:13:32.331052   20774 configmap.go:190] 33:33:33::33::33 is not a valid textual representation of an IP address
W0704 13:13:32.334582   20774 configmap.go:150] demo is not a valid http code: strconv.Atoi: parsing "demo": invalid syntax
W0704 13:13:32.334625   20774 configmap.go:190] 3.3.3 is not a valid textual representation of an IP address
W0704 13:13:32.334636   20774 configmap.go:190] 33:33:33::33::33 is not a valid textual representation of an IP address
--- PASS: TestMergeConfigMapToStruct (0.01s)
=== RUN   TestGlobalExternalAuthURLParsing
W0704 13:13:32.339731   20774 configmap.go:230] Global auth location denied - url scheme is empty.
W0704 13:13:32.342217   20774 configmap.go:230] Global auth location denied - url host is empty.
W0704 13:13:32.343262   20774 configmap.go:230] Global auth location denied - invalid url host.
--- PASS: TestGlobalExternalAuthURLParsing (0.01s)
=== RUN   TestGlobalExternalAuthMethodParsing
W0704 13:13:32.345558   20774 configmap.go:242] Global auth location denied - invalid HTTP method.
--- PASS: TestGlobalExternalAuthMethodParsing (0.00s)
=== RUN   TestGlobalExternalAuthSigninParsing
W0704 13:13:32.348132   20774 configmap.go:254] Global auth location denied - global-auth-signin setting is undefined and will not be set.
W0704 13:13:32.349570   20774 configmap.go:254] Global auth location denied - global-auth-signin setting is undefined and will not be set.
W0704 13:13:32.351214   20774 configmap.go:254] Global auth location denied - global-auth-signin setting is undefined and will not be set.
--- PASS: TestGlobalExternalAuthSigninParsing (0.01s)
=== RUN   TestGlobalExternalAuthSigninRedirectParamParsing
W0704 13:13:32.354785   20774 configmap.go:254] Global auth location denied - global-auth-signin setting is undefined and will not be set.
W0704 13:13:32.354804   20774 configmap.go:267] Global auth redirect parameter denied - global-auth-signin-redirect-param setting is invalid and will not be set.
--- PASS: TestGlobalExternalAuthSigninRedirectParamParsing (0.00s)
=== RUN   TestGlobalExternalAuthResponseHeadersParsing
W0704 13:13:32.358287   20774 configmap.go:281] Global auth location denied - invalid headers list.
W0704 13:13:32.359164   20774 configmap.go:281] Global auth location denied - invalid headers list.
W0704 13:13:32.362364   20774 configmap.go:281] Global auth location denied - invalid headers list.
--- PASS: TestGlobalExternalAuthResponseHeadersParsing (0.01s)
=== RUN   TestGlobalExternalAuthRequestRedirectParsing
--- PASS: TestGlobalExternalAuthRequestRedirectParsing (0.00s)
=== RUN   TestGlobalExternalAuthSnippetParsing
--- PASS: TestGlobalExternalAuthSnippetParsing (0.00s)
=== RUN   TestGlobalExternalAuthCacheDurationParsing
W0704 13:13:32.367914   20774 configmap.go:311] Global auth location denied - Location denied, reason: invalid cache duration: 200
W0704 13:13:32.369124   20774 configmap.go:311] Global auth location denied - Location denied, reason: invalid cache duration: xaxax
--- PASS: TestGlobalExternalAuthCacheDurationParsing (0.01s)
=== RUN   TestLuaSharedDictsParsing
E0704 13:13:32.378121   20774 configmap.go:122] Ignoring non integer value 1a for Lua dictionary invalid_dict: strconv.Atoi: parsing "1a": invalid syntax.
E0704 13:13:32.378900   20774 configmap.go:126] Ignoring 201 for Lua dictionary invalid_dict: maximum size is 200.
--- PASS: TestLuaSharedDictsParsing (0.00s)
=== RUN   TestSplitAndTrimSpace
--- PASS: TestSplitAndTrimSpace (0.00s)
=== RUN   TestBuildLuaSharedDictionaries
E0704 13:13:32.379661   20774 template.go:329] expected an '[]*ingress.Server' type but *ingress.Ingress was returned
E0704 13:13:32.379680   20774 template.go:323] expected a 'config.Configuration' type but *ingress.Ingress was returned
--- PASS: TestBuildLuaSharedDictionaries (0.00s)
=== RUN   TestLuaConfigurationRequestBodySize
--- PASS: TestLuaConfigurationRequestBodySize (0.00s)
=== RUN   TestFormatIP
--- PASS: TestFormatIP (0.00s)
=== RUN   TestQuote
--- PASS: TestQuote (0.00s)
=== RUN   TestBuildLocation
E0704 13:13:32.379884   20774 template.go:514] expected an '*ingress.Location' type but *ingress.Ingress was returned
--- PASS: TestBuildLocation (0.00s)
=== RUN   TestBuildProxyPass
--- PASS: TestBuildProxyPass (0.00s)
=== RUN   TestBuildAuthLocation
E0704 13:13:32.380068   20774 template.go:533] expected an '*ingress.Location' type but *ingress.Ingress was returned
--- PASS: TestBuildAuthLocation (0.00s)
=== RUN   TestShouldApplyGlobalAuth
--- PASS: TestShouldApplyGlobalAuth (0.00s)
=== RUN   TestBuildAuthResponseHeaders
--- PASS: TestBuildAuthResponseHeaders (0.00s)
=== RUN   TestBuildAuthProxySetHeaders
--- PASS: TestBuildAuthProxySetHeaders (0.00s)
=== RUN   TestTemplateWithData
--- PASS: TestTemplateWithData (0.19s)
=== RUN   TestBuildDenyVariable
E0704 13:13:32.572839   20774 template.go:843] expected a 'string' type but *ingress.Ingress was returned
--- PASS: TestBuildDenyVariable (0.00s)
=== RUN   TestBuildByteSize
E0704 13:13:32.572926   20774 template.go:906] expected an 'string' type but <nil> was returned
--- PASS: TestBuildByteSize (0.00s)
=== RUN   TestIsLocationAllowed
E0704 13:13:32.572974   20774 template.go:824] expected an '*ingress.Location' type but *ingress.Ingress was returned
--- PASS: TestIsLocationAllowed (0.00s)
=== RUN   TestBuildForwardedFor
E0704 13:13:32.573004   20774 template.go:1037] expected a 'string' type but *ingress.Ingress was returned
--- PASS: TestBuildForwardedFor (0.00s)
=== RUN   TestBuildResolvers
E0704 13:13:32.573048   20774 template.go:452] expected a '[]net.IP' type but *ingress.Ingress was returned
E0704 13:13:32.573052   20774 template.go:457] expected a 'bool' type but *ingress.Ingress was returned
--- PASS: TestBuildResolvers (0.00s)
=== RUN   TestBuildNextUpstream
E0704 13:13:32.573087   20774 template.go:869] expected a 'string' type but *ingress.Ingress was returned
--- PASS: TestBuildNextUpstream (0.00s)
=== RUN   TestBuildRateLimit
E0704 13:13:32.573125   20774 template.go:762] expected an '*ingress.Location' type but *ingress.Ingress was returned
E0704 13:13:32.573142   20774 template.go:762] expected an '*ingress.Location' type but *ingress.Ingress was returned
--- PASS: TestBuildRateLimit (0.00s)
=== RUN   TestBuildRateLimitZones
E0704 13:13:32.573185   20774 template.go:712] expected a '[]*ingress.Server' type but *ingress.Ingress was returned
--- PASS: TestBuildRateLimitZones (0.00s)
=== RUN   TestFilterRateLimits
E0704 13:13:32.573213   20774 template.go:689] expected a '[]ratelimit.RateLimit' type but *ingress.Ingress was returned
--- PASS: TestFilterRateLimits (0.00s)
=== RUN   TestBuildAuthSignURL
--- PASS: TestBuildAuthSignURL (0.00s)
=== RUN   TestIsLocationInLocationList
E0704 13:13:32.573301   20774 template.go:802] expected an '*ingress.Location' type but *ingress.Ingress was returned
--- PASS: TestIsLocationInLocationList (0.00s)
=== RUN   TestBuildUpstreamName
E0704 13:13:32.573334   20774 template.go:857] expected a '*ingress.Location' type but *ingress.Ingress was returned
--- PASS: TestBuildUpstreamName (0.00s)
=== RUN   TestEscapeLiteralDollar
--- PASS: TestEscapeLiteralDollar (0.00s)
=== RUN   TestOpentracingPropagateContext
--- PASS: TestOpentracingPropagateContext (0.00s)
=== RUN   TestGetIngressInformation
E0704 13:13:32.573465   20774 template.go:955] expected an '*ingress.Ingress' type but string was returned
E0704 13:13:32.573477   20774 template.go:967] expected a 'string' type but int was returned
--- PASS: TestGetIngressInformation (0.00s)
=== RUN   TestBuildCustomErrorLocationsPerServer
--- PASS: TestBuildCustomErrorLocationsPerServer (0.00s)
=== RUN   TestProxySetHeader
E0704 13:13:32.573592   20774 template.go:1150] expected a '*ingress.Location' type but *ingress.Ingress was returned
--- PASS: TestProxySetHeader (0.00s)
=== RUN   TestBuildInfluxDB
E0704 13:13:32.573632   20774 template.go:1130] expected an 'influxdb.Config' type but *ingress.Ingress was returned
--- PASS: TestBuildInfluxDB (0.00s)
=== RUN   TestBuildOpenTracing
E0704 13:13:32.573675   20774 template.go:1088] expected a 'config.Configuration' type but *ingress.Ingress was returned
--- PASS: TestBuildOpenTracing (0.00s)
=== RUN   TestEnforceRegexModifier
E0704 13:13:32.573732   20774 template.go:497] expected an '[]*ingress.Location' type but *ingress.Ingress was returned
--- PASS: TestEnforceRegexModifier (0.00s)
=== RUN   TestShouldLoadModSecurityModule
E0704 13:13:32.573765   20774 template.go:1255] expected an '[]*ingress.Server' type but *ingress.Ingress was returned
E0704 13:13:32.573767   20774 template.go:1249] expected a 'config.Configuration' type but *ingress.Ingress was returned
--- PASS: TestShouldLoadModSecurityModule (0.00s)
=== RUN   TestOpentracingForLocation
--- PASS: TestOpentracingForLocation (0.00s)
=== RUN   TestShouldLoadOpentracingModule
E0704 13:13:32.573827   20774 template.go:1474] expected an '[]*ingress.Server' type but *ingress.Ingress was returned
E0704 13:13:32.573829   20774 template.go:1468] expected a 'config.Configuration' type but *ingress.Ingress was returned
--- PASS: TestShouldLoadOpentracingModule (0.00s)
=== RUN   TestModSecurityForLocation
--- PASS: TestModSecurityForLocation (0.00s)
=== RUN   TestBuildServerName
--- PASS: TestBuildServerName (0.00s)
=== RUN   TestParseComplexNginxVarIntoLuaTable
--- PASS: TestParseComplexNginxVarIntoLuaTable (0.00s)
=== RUN   TestConvertGoSliceIntoLuaTablet
--- PASS: TestConvertGoSliceIntoLuaTablet (0.00s)
=== RUN   TestCleanConf
--- PASS: TestCleanConf (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/controller/template       (cached)
?       k8s.io/ingress-nginx/internal/ingress/defaults  [no test files]
=== RUN   TestIsLocationDenied
--- PASS: TestIsLocationDenied (0.00s)
=== RUN   TestIsMissingAnnotations
--- PASS: TestIsMissingAnnotations (0.00s)
=== RUN   TestInvalidContent
--- PASS: TestInvalidContent (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/errors    (cached)
?       k8s.io/ingress-nginx/internal/ingress/metric    [no test files]
=== RUN   TestControllerCounters
=== RUN   TestControllerCounters/should_return_not_increment_in_metrics_if_no_operations_are_invoked
=== RUN   TestControllerCounters/single_increase_in_reload_count_should_return_1
=== RUN   TestControllerCounters/single_increase_in_error_reload_count_should_return_1
=== RUN   TestControllerCounters/should_set_SSL_certificates_metrics
--- PASS: TestControllerCounters (0.00s)
    --- PASS: TestControllerCounters/should_return_not_increment_in_metrics_if_no_operations_are_invoked (0.00s)
    --- PASS: TestControllerCounters/single_increase_in_reload_count_should_return_1 (0.00s)
    --- PASS: TestControllerCounters/single_increase_in_error_reload_count_should_return_1 (0.00s)
    --- PASS: TestControllerCounters/should_set_SSL_certificates_metrics (0.00s)
=== RUN   TestRemoveMetrics
--- PASS: TestRemoveMetrics (0.00s)
=== RUN   TestStatusCollector
=== RUN   TestStatusCollector/should_return_empty_metrics
=== RUN   TestStatusCollector/should_return_metrics_for_total_connections
=== RUN   TestStatusCollector/should_return_nginx_metrics_all_available_metrics
--- PASS: TestStatusCollector (3.00s)
    --- PASS: TestStatusCollector/should_return_empty_metrics (1.00s)
    --- PASS: TestStatusCollector/should_return_metrics_for_total_connections (1.00s)
    --- PASS: TestStatusCollector/should_return_nginx_metrics_all_available_metrics (1.00s)
=== RUN   TestProcessCollector
=== RUN   TestProcessCollector/should_return_metrics
    process_test.go:54: Signal: killed
--- PASS: TestProcessCollector (0.03s)
    --- PASS: TestProcessCollector/should_return_metrics (0.03s)
=== RUN   TestNewUDPLogListener
--- PASS: TestNewUDPLogListener (0.00s)
=== RUN   TestCollector
=== RUN   TestCollector/invalid_metric_object_should_not_increase_prometheus_metrics
E0704 13:13:35.351763   20746 socket.go:228] "Unexpected error deserializing JSON" err="[]collectors.socketData: decode slice: expect [ or n, but found #, error found in #1 byte of ...|#missing {\n|..., bigger context ...|#missing {\n\t\t\t\t\"host\":\"testshop.com\",\n\t\t\t\t\"status\":|..." payload="#missing {\n\t\t\t\t\"host\":\"testshop.com\",\n\t\t\t\t\"status\":\"200\",\n\t\t\t\t\"bytesSent\":150.0,\n\t\t\t\t\"method\":\"GET\",\n\t\t\t\t\"path\":\"/admin\",\n\t\t\t\t\"requestLength\":300.0,\n\t\t\t\t\"requestTime\":60.0,\n\t\t\t\t\"upstreamName\":\"test-upstream\",\n\t\t\t\t\"upstreamIP\":\"1.1.1.1:8080\",\n\t\t\t\t\"upstreamResponseTime\":200,\n\t\t\t\t\"upstreamStatus\":\"220\",\n\t\t\t\t\"namespace\":\"test-app-production\",\n\t\t\t\t\"ingress\":\"web-yml\",\n\t\t\t\t\"service\":\"test-app\"\n\t\t\t}"
=== RUN   TestCollector/valid_metric_object_should_update_prometheus_metrics
=== RUN   TestCollector/multiple_messages_should_increase_prometheus_metric_by_two
=== RUN   TestCollector/collector_should_be_able_to_handle_batched_metrics_correctly
--- PASS: TestCollector (2.00s)
    --- PASS: TestCollector/invalid_metric_object_should_not_increase_prometheus_metrics (0.00s)
    --- PASS: TestCollector/valid_metric_object_should_update_prometheus_metrics (1.00s)
    --- PASS: TestCollector/multiple_messages_should_increase_prometheus_metric_by_two (0.00s)
    --- PASS: TestCollector/collector_should_be_able_to_handle_batched_metrics_correctly (1.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/metric/collectors (cached)
=== RUN   TestAuthSSLCertEqual
--- PASS: TestAuthSSLCertEqual (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/resolver  (cached)
=== RUN   TestStatusActions
I0704 13:13:32.664661   20874 status.go:284] "updating Ingress status" namespace="default" ingress="foo_ingress_1" currentValue=[{IP:10.0.0.1 Hostname:foo1 Ports:[]}] newValue=[{IP:11.0.0.2 Hostname: Ports:[]}]
I0704 13:13:32.766657   20874 status.go:284] "updating Ingress status" namespace="default" ingress="foo_ingress_1" currentValue=[{IP:11.0.0.2 Hostname: Ports:[]}] newValue=[{IP:11.0.0.2 Hostname: Ports:[]}]
I0704 13:13:33.766996   20874 status.go:132] "removing value from ingress status" address=[11.0.0.2]
I0704 13:13:33.767093   20874 status.go:284] "updating Ingress status" namespace="default" ingress="foo_ingress_1" currentValue=[{IP:11.0.0.2 Hostname: Ports:[]}] newValue=[]
--- PASS: TestStatusActions (1.10s)
=== RUN   TestCallback
--- PASS: TestCallback (0.00s)
=== RUN   TestKeyfunc
--- PASS: TestKeyfunc (0.00s)
=== RUN   TestRunningAddressesWithPublishService
=== RUN   TestRunningAddressesWithPublishService/service_type_ClusterIP
=== RUN   TestRunningAddressesWithPublishService/service_type_NodePort
=== RUN   TestRunningAddressesWithPublishService/service_type_ExternalName
=== RUN   TestRunningAddressesWithPublishService/service_type_LoadBalancer
=== RUN   TestRunningAddressesWithPublishService/service_type_LoadBalancer_with_same_externalIP_and_ingress_IP
=== RUN   TestRunningAddressesWithPublishService/invalid_service_type
--- PASS: TestRunningAddressesWithPublishService (0.00s)
    --- PASS: TestRunningAddressesWithPublishService/service_type_ClusterIP (0.00s)
    --- PASS: TestRunningAddressesWithPublishService/service_type_NodePort (0.00s)
    --- PASS: TestRunningAddressesWithPublishService/service_type_ExternalName (0.00s)
    --- PASS: TestRunningAddressesWithPublishService/service_type_LoadBalancer (0.00s)
    --- PASS: TestRunningAddressesWithPublishService/service_type_LoadBalancer_with_same_externalIP_and_ingress_IP (0.00s)
    --- PASS: TestRunningAddressesWithPublishService/invalid_service_type (0.00s)
=== RUN   TestRunningAddressesWithPods
--- PASS: TestRunningAddressesWithPods (0.00s)
=== RUN   TestRunningAddressesWithPublishStatusAddress
--- PASS: TestRunningAddressesWithPublishStatusAddress (0.00s)
=== RUN   TestRunningAddressesWithPublishStatusAddresses
--- PASS: TestRunningAddressesWithPublishStatusAddresses (0.00s)
=== RUN   TestRunningAddressesWithPublishStatusAddressesAndSpaces
--- PASS: TestRunningAddressesWithPublishStatusAddressesAndSpaces (0.00s)
=== RUN   TestSliceToStatus
--- PASS: TestSliceToStatus (0.00s)
=== RUN   TestIngressSliceEqual
--- PASS: TestIngressSliceEqual (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/ingress/status    (cached)
=== RUN   TestParseNameNS
--- PASS: TestParseNameNS (0.00s)
=== RUN   TestGetNodeIP
E0704 13:13:32.476580   20859 main.go:49] "Error getting node" err="nodes \"demo\" not found" name="demo"
E0704 13:13:32.476609   20859 main.go:49] "Error getting node" err="nodes \"notexistnode\" not found" name="notexistnode"
--- PASS: TestGetNodeIP (0.00s)
=== RUN   TestGetIngressPod
--- PASS: TestGetIngressPod (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/k8s       (cached)
=== RUN   TestNewIPSet
--- PASS: TestNewIPSet (0.00s)
=== RUN   TestParseCIDRs
--- PASS: TestParseCIDRs (0.00s)
=== RUN   TestIsIPV6
--- PASS: TestIsIPV6 (0.00s)
=== RUN   TestIsPortAvailable
--- PASS: TestIsPortAvailable (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/net       (cached)
=== RUN   TestGetDNSServers
--- PASS: TestGetDNSServers (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/net/dns   (cached)
=== RUN   TestStoreSSLCertOnDisk
    ssl_test.go:90: unexpected error storing SSL certificate: could not create PEM certificate file /etc/ingress-controller/ssl/test-1625409008081901400.pem: open /etc/ingress-controller/ssl/test-1625409008081901400.pem: no such file or directory
--- FAIL: TestStoreSSLCertOnDisk (0.18s)
=== RUN   TestCACert
    ssl_test.go:125: unexpected error storing SSL certificate: could not create PEM certificate file /etc/ingress-controller/ssl/test-1625409008438239700.pem: open /etc/ingress-controller/ssl/test-1625409008438239700.pem: no such file or directory
--- FAIL: TestCACert (0.36s)
=== RUN   TestGetFakeSSLCert
F0704 14:30:08.540042    1895 ssl.go:389] unexpected error storing fake SSL Cert: could not create PEM certificate file /etc/ingress-controller/ssl/default-fake-certificate.pem: open /etc/ingress-controller/ssl/default-fake-certificate.pem: no such file or directory
goroutine 50 [running]:
k8s.io/klog/v2.stacks(0xc000126001, 0xc0001526e0, 0x10b, 0x160)
        /go/pkg/mod/k8s.io/klog/v2@v2.8.0/klog.go:1021 +0xb9
k8s.io/klog/v2.(*loggingT).output(0x1ef27a0, 0xc000000003, 0x0, 0x0, 0xc0003b61c0, 0x1e614f1, 0x6, 0x185, 0x0)
        /go/pkg/mod/k8s.io/klog/v2@v2.8.0/klog.go:970 +0x19b
k8s.io/klog/v2.(*loggingT).printf(0x1ef27a0, 0xc000000003, 0x0, 0x0, 0x0, 0x0, 0x151a4ec, 0x2a, 0xc00038cfa0, 0x1, ...)
        /go/pkg/mod/k8s.io/klog/v2@v2.8.0/klog.go:751 +0x191
k8s.io/klog/v2.Fatalf(...)
        /go/pkg/mod/k8s.io/klog/v2@v2.8.0/klog.go:1509
k8s.io/ingress-nginx/internal/net/ssl.GetFakeSSLCert(0x174405f376fe8)
        /go/src/k8s.io/ingress-nginx/internal/net/ssl/ssl.go:389 +0x28d
k8s.io/ingress-nginx/internal/net/ssl.TestGetFakeSSLCert(0xc000503800)
        /go/src/k8s.io/ingress-nginx/internal/net/ssl/ssl_test.go:141 +0x26
testing.tRunner(0xc000503800, 0x159d9b8)
        /usr/local/go/src/testing/testing.go:1123 +0xef
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:1168 +0x2b3

goroutine 1 [chan receive]:
testing.(*T).Run(0xc000503800, 0x14fef6b, 0x12, 0x159d9b8, 0x499600)
        /usr/local/go/src/testing/testing.go:1169 +0x2da
testing.runTests.func1(0xc0003a4180)
        /usr/local/go/src/testing/testing.go:1439 +0x78
testing.tRunner(0xc0003a4180, 0xc000179de0)
        /usr/local/go/src/testing/testing.go:1123 +0xef
testing.runTests(0xc00039e660, 0x1edf840, 0x8, 0x8, 0xc0308ff1f5c3ba74, 0x8bb2fdffdd, 0x1ef2460, 0x40f690)
        /usr/local/go/src/testing/testing.go:1437 +0x2fe
testing.(*M).Run(0xc000374a80, 0x0)
        /usr/local/go/src/testing/testing.go:1345 +0x1eb
main.main()
        _testmain.go:57 +0x138

goroutine 19 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x1ef27a0)
        /go/pkg/mod/k8s.io/klog/v2@v2.8.0/klog.go:1164 +0x8b
created by k8s.io/klog/v2.init.0
        /go/pkg/mod/k8s.io/klog/v2@v2.8.0/klog.go:418 +0xdf
FAIL    k8s.io/ingress-nginx/internal/net/ssl   0.645s
?       k8s.io/ingress-nginx/internal/nginx     [no test files]
?       k8s.io/ingress-nginx/internal/runtime   [no test files]
=== RUN   TestElementsMatch
--- PASS: TestElementsMatch (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/sets      (cached)
=== RUN   TestShutdown
--- PASS: TestShutdown (0.00s)
=== RUN   TestEnqueueSuccess
--- PASS: TestEnqueueSuccess (0.01s)
=== RUN   TestEnqueueFailed
E0704 13:13:23.267789   11704 queue.go:78] "queue has been shutdown, failed to enqueue" key={k:testKey v:testValue}
--- PASS: TestEnqueueFailed (0.02s)
=== RUN   TestEnqueueKeyError
E0704 13:13:23.278019   11704 queue.go:90] "creating object key" err="failed to get key" item={k:testKey v:testValue}
--- PASS: TestEnqueueKeyError (0.01s)
=== RUN   TestSkipEnqueue
--- PASS: TestSkipEnqueue (0.01s)
PASS
ok      k8s.io/ingress-nginx/internal/task      (cached)
=== RUN   TestFileWatcher
--- PASS: TestFileWatcher (0.50s)
=== RUN   TestFileWatcherWithNestedSymlink
--- PASS: TestFileWatcherWithNestedSymlink (0.00s)
PASS
ok      k8s.io/ingress-nginx/internal/watch     (cached)
?       k8s.io/ingress-nginx/version    [no test files]
FAIL
make: *** [Makefile:111: test] Error 1
```

</details>

<details>
<summary>e2e test logs</summary>

```
make kind-e2e-test
[dev-env] creating Kubernetes cluster with kind
Creating cluster "ingress-nginx-dev" ...
DEBUG: docker/images.go:58] Image: kindest/node:v1.20.2@sha256:8f7ea6e7642c0da54f04a7ee10431549c0257315b3a634f6ef2fecaaedb19bab present locally
 ✓ Ensuring node image (kindest/node:v1.20.2) 🖼
 ✓ Preparing nodes 📦 📦 📦
 ✓ Writing configuration 📜
 ✓ Starting control-plane 🕹️
 ✓ Installing CNI 🔌
 ✓ Installing StorageClass 💾
 ✓ Joining worker nodes 🚜
Set kubectl context to "kind-ingress-nginx-dev"
You can now use your cluster with:

kubectl cluster-info --context kind-ingress-nginx-dev

Have a question, bug, or feature request? Let us know! https://kind.sigs.k8s.io/#community 🙂
Kubernetes cluster:
NAME                              STATUS     ROLES                  AGE   VERSION   INTERNAL-IP   EXTERNAL-IP   OS-IMAGE       KERNEL-VERSION                CONTAINER-RUNTIME
ingress-nginx-dev-control-plane   Ready      control-plane,master   33s   v1.20.2   172.19.0.4    <none>        Ubuntu 20.10   4.19.128-microsoft-standard   containerd://1.4.0-106-gce4439a8
ingress-nginx-dev-worker          NotReady   <none>                 2s    v1.20.2   172.19.0.2    <none>        Ubuntu 20.10   4.19.128-microsoft-standard   containerd://1.4.0-106-gce4439a8
ingress-nginx-dev-worker2         NotReady   <none>                 2s    v1.20.2   172.19.0.3    <none>        Ubuntu 20.10   4.19.128-microsoft-standard   containerd://1.4.0-106-gce4439a8
go: downloading github.com/onsi/ginkgo v1.16.4
go: downloading github.com/go-task/slim-sprig v0.0.0-20210107165309-348f09dbbbc0
go: downloading github.com/nxadm/tail v1.4.8
go get: upgraded github.com/onsi/ginkgo v1.14.1 => v1.16.4
[dev-env] building image
removing old image ingress-controller/controller:1.0.0-dev
Error: No such image: ingress-controller/controller:1.0.0-dev
go: downloading k8s.io/apiserver v0.21.1
go: downloading k8s.io/apimachinery v0.21.1
go: downloading k8s.io/client-go v0.21.1
go: downloading github.com/spf13/pflag v1.0.5
go: downloading github.com/pkg/errors v0.9.1
go: downloading k8s.io/klog/v2 v2.8.0
go: downloading k8s.io/api v0.21.1
go: downloading github.com/go-logr/logr v0.4.0
go: downloading github.com/mitchellh/copystructure v1.0.0
go: downloading github.com/eapache/channels v1.1.0
go: downloading github.com/imdario/mergo v0.3.10
go: downloading github.com/prometheus/client_golang v1.7.1
go: downloading github.com/mitchellh/reflectwalk v1.0.0
go: downloading github.com/fsnotify/fsnotify v1.4.9
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/armon/go-proxyproto v0.0.0-20200108142055-f0b8253b1507
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading golang.org/x/sys v0.0.0-20210225134936-a50acf3fe073
go: downloading sigs.k8s.io/structured-merge-diff/v4 v4.1.0
go: downloading k8s.io/utils v0.0.0-20201110183641-67b214c5f920
go: downloading golang.org/x/net v0.0.0-20210224082022-3d97a244fca7
go: downloading gopkg.in/go-playground/pool.v3 v3.1.1
go: downloading golang.org/x/term v0.0.0-20210220032956-6a3ed077a48d
go: downloading github.com/ncabatoff/process-exporter v0.7.2
go: downloading github.com/json-iterator/go v1.1.10
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading pault.ag/go/sniff v0.0.0-20200207005214-cf7e4d167732
go: downloading github.com/prometheus/common v0.14.0
go: downloading github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
go: downloading github.com/mitchellh/hashstructure v1.0.0
go: downloading github.com/zakjan/cert-chain-resolver v0.0.0-20200729110141-6b99e360f97a
go: downloading github.com/google/go-cmp v0.5.2
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/google/gofuzz v1.1.0
go: downloading github.com/prometheus/procfs v0.2.0
go: downloading github.com/golang/protobuf v1.4.3
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/mitchellh/mapstructure v1.3.2
go: downloading github.com/mitchellh/go-ps v1.0.0
go: downloading github.com/ncabatoff/go-seq v0.0.0-20180805175032-b08ef85ed833
go: downloading github.com/beorn7/perks v1.0.1
go: downloading golang.org/x/text v0.3.4
go: downloading github.com/fullsailor/pkcs7 v0.0.0-20190404230743-d7302db945fa
go: downloading github.com/hashicorp/golang-lru v0.5.4
go: downloading github.com/googleapis/gnostic v0.5.1
go: downloading golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369
go: downloading google.golang.org/protobuf v1.25.0
go: downloading sigs.k8s.io/yaml v1.2.0
go: downloading github.com/opencontainers/runc v1.0.0-rc92
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading golang.org/x/time v0.0.0-20210220033141-f8bda1e9f3ba
go: downloading github.com/cespare/xxhash v1.1.0
go: downloading k8s.io/kube-openapi v0.0.0-20210305001622-591a79e4bda7
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/docker/go-units v0.4.0
go: downloading github.com/cyphar/filepath-securejoin v0.2.2
go: downloading k8s.io/component-base v0.21.1
go: downloading github.com/google/uuid v1.1.2
go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
go: downloading github.com/sirupsen/logrus v1.7.0
go: downloading github.com/opencontainers/runtime-spec v1.0.3-0.20200728170252-4d89ac9fbff6
go: downloading github.com/emicklei/go-restful v2.9.5+incompatible
go: downloading github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e
go: downloading github.com/blang/semver v3.5.1+incompatible
go: downloading github.com/coreos/go-systemd/v22 v22.1.0
go: downloading github.com/godbus/dbus/v5 v5.0.3
go: downloading github.com/spf13/cobra v1.1.1
Building docker image (amd64)...
[+] Building 3.7s (15/15) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                                                                   0.0s
 => => transferring dockerfile: 38B                                                                                                                                                                                                    0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                      0.0s
 => => transferring context: 34B                                                                                                                                                                                                       0.0s
 => [internal] load metadata for k8s.gcr.io/ingress-nginx/nginx:v20210530-g6aab4c291@sha256:a7356029dd0c26cc3466bf7a27daec0f4df73aa14ca6c8b871a767022a812c0b                                                                           0.3s
 => [ 1/10] FROM k8s.gcr.io/ingress-nginx/nginx:v20210530-g6aab4c291@sha256:a7356029dd0c26cc3466bf7a27daec0f4df73aa14ca6c8b871a767022a812c0b                                                                                           0.0s
 => [internal] load build context                                                                                                                                                                                                      0.3s
 => => transferring context: 49.46MB                                                                                                                                                                                                   0.3s
 => CACHED [ 2/10] WORKDIR  /etc/nginx                                                                                                                                                                                                 0.0s
 => [ 3/10] RUN apk update   && apk upgrade   && apk add --no-cache     diffutils   && rm -rf /var/cache/apk/*                                                                                                                         1.5s
 => [ 4/10] COPY --chown=www-data:www-data etc /etc                                                                                                                                                                                    0.0s
 => [ 5/10] COPY --chown=www-data:www-data bin/amd64/dbg /                                                                                                                                                                             0.0s
 => [ 6/10] COPY --chown=www-data:www-data bin/amd64/nginx-ingress-controller /                                                                                                                                                        0.1s
 => [ 7/10] COPY --chown=www-data:www-data bin/amd64/wait-shutdown /                                                                                                                                                                   0.0s
 => [ 8/10] RUN bash -xeu -c '   writeDirs=(     /etc/ingress-controller     /etc/ingress-controller/ssl     /etc/ingress-controller/auth     /var/log     /var/log/nginx   );   for dir in "${writeDirs[@]}"; do     mkdir -p ${dir}  0.3s
 => [ 9/10] RUN apk add --no-cache libcap   && setcap    cap_net_bind_service=+ep /nginx-ingress-controller   && setcap -v cap_net_bind_service=+ep /nginx-ingress-controller   && setcap    cap_net_bind_service=+ep /usr/local/ngin  0.8s
 => [10/10] RUN  ln -sf /dev/stdout /var/log/nginx/access.log   && ln -sf /dev/stderr /var/log/nginx/error.log                                                                                                                         0.3s
 => exporting to image                                                                                                                                                                                                                 0.2s
 => => exporting layers                                                                                                                                                                                                                0.2s
 => => writing image sha256:c64ba05221ebe3e006eb78d86e2faef1a3ebf74d5023e9dc5119530e1fa11d78                                                                                                                                           0.0s
 => => naming to docker.io/ingress-controller/controller:1.0.0-dev                                                                                                                                                                     0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
Compiling e2e...
go: downloading k8s.io/apimachinery v0.21.1
go: downloading k8s.io/api v0.21.1
go: downloading github.com/gavv/httpexpect/v2 v2.1.0
go: downloading github.com/onsi/ginkgo v1.16.4
go: downloading k8s.io/client-go v0.21.1
go: downloading github.com/stretchr/testify v1.6.1
go: downloading golang.org/x/crypto v0.0.0-20210220033148-5ea612d1eb83
go: downloading github.com/valyala/fasthttp v1.9.0
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading github.com/mitchellh/go-ps v1.0.0
go: downloading github.com/fatih/structs v1.0.0
go: downloading k8s.io/component-base v0.21.1
go: downloading github.com/google/uuid v1.1.2
go: downloading github.com/valyala/bytebufferpool v1.0.0
go: downloading golang.org/x/net v0.0.0-20210224082022-3d97a244fca7
go: downloading moul.io/http2curl v1.0.1-0.20190925090545-5cd742060b0e
go: downloading github.com/yalp/jsonpath v0.0.0-20180802001716-5cc68e5049a0
go: downloading github.com/prometheus/common v0.14.0
go: downloading github.com/gorilla/websocket v1.4.2
go: downloading github.com/go-logr/logr v0.4.0
go: downloading github.com/klauspost/compress v1.8.2
go: downloading github.com/moul/pb v0.0.0-20180404114147-54bdd96e6a52
go: downloading github.com/google/gofuzz v1.1.0
go: downloading github.com/yudai/gojsondiff v1.0.0
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/sergi/go-diff v1.1.0
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/golang/protobuf v1.4.3
go: downloading github.com/spf13/pflag v1.0.5
go: downloading golang.org/x/term v0.0.0-20210220032956-6a3ed077a48d
go: downloading golang.org/x/text v0.3.4
go: downloading golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading gopkg.in/go-playground/pool.v3 v3.1.1
go: downloading golang.org/x/sys v0.0.0-20210225134936-a50acf3fe073
go: downloading github.com/xeipuuv/gojsonschema v1.1.0
go: downloading google.golang.org/protobuf v1.25.0
go: downloading github.com/Azure/go-autorest v14.2.0+incompatible
go: downloading github.com/Azure/go-autorest/autorest v0.11.12
go: downloading github.com/Azure/go-autorest/autorest/adal v0.9.5
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading github.com/google/go-querystring v1.0.0
go: downloading sigs.k8s.io/structured-merge-diff/v4 v4.1.0
go: downloading github.com/Azure/go-autorest/tracing v0.6.0
go: downloading github.com/imdario/mergo v0.3.10
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading k8s.io/klog/v2 v2.8.0
go: downloading github.com/ajg/form v1.5.1
go: downloading github.com/Azure/go-autorest/autorest/date v0.3.0
go: downloading github.com/imkira/go-interpol v1.0.0
go: downloading google.golang.org/grpc v1.27.1
go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
go: downloading github.com/xeipuuv/gojsonreference v0.0.0-20180127040603-bd5ef7bd5415
go: downloading k8s.io/utils v0.0.0-20201110183641-67b214c5f920
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/form3tech-oss/jwt-go v3.2.2+incompatible
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369
go: downloading go.uber.org/zap v1.15.0
go: downloading github.com/yudai/golcs v0.0.0-20170316035057-ecda9a501e82
go: downloading sigs.k8s.io/yaml v1.2.0
go: downloading cloud.google.com/go v0.54.0
go: downloading github.com/google/go-cmp v0.5.2
go: downloading github.com/Azure/go-autorest/logger v0.2.0
go: downloading go.uber.org/atomic v1.6.0
go: downloading go.uber.org/multierr v1.5.0
go: downloading github.com/xeipuuv/gojsonpointer v0.0.0-20180127040702-4e3ac2762d5f
go: downloading github.com/nxadm/tail v1.4.8
go: downloading github.com/json-iterator/go v1.1.10
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading k8s.io/apiextensions-apiserver v0.21.1
go: downloading github.com/fsnotify/fsnotify v1.4.9
go: downloading gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7
go: downloading github.com/hashicorp/golang-lru v0.5.4
go: downloading golang.org/x/time v0.0.0-20210220033141-f8bda1e9f3ba
go: downloading google.golang.org/genproto v0.0.0-20201110150050-8816d57aaa9a
go: downloading github.com/googleapis/gnostic v0.5.1
go: downloading github.com/klauspost/cpuid v1.2.1

    compiled e2e.test
[+] Building 4.1s (15/15) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                                                                   0.0s
 => => transferring dockerfile: 614B                                                                                                                                                                                                   0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                      0.0s
 => => transferring context: 2B                                                                                                                                                                                                        0.0s
 => [internal] load metadata for docker.io/library/alpine:3.12                                                                                                                                                                         2.2s
 => [internal] load metadata for k8s.gcr.io/ingress-nginx/e2e-test-runner:v20210601-g96a87c79b@sha256:f84dcddc84e5cba220260f315e18cd47fc8c6b7f3f4f57b7b3e9cc2ea25324b7                                                                 0.0s
 => [base 1/1] FROM k8s.gcr.io/ingress-nginx/e2e-test-runner:v20210601-g96a87c79b@sha256:f84dcddc84e5cba220260f315e18cd47fc8c6b7f3f4f57b7b3e9cc2ea25324b7                                                                              0.5s
 => [internal] load build context                                                                                                                                                                                                      0.5s
 => => transferring context: 55.74MB                                                                                                                                                                                                   0.4s
 => [stage-1 1/8] FROM docker.io/library/alpine:3.12@sha256:87703314048c40236c6d674424159ee862e2b96ce1c37c62d877e21ed27a387e                                                                                                           0.0s
 => => resolve docker.io/library/alpine:3.12@sha256:87703314048c40236c6d674424159ee862e2b96ce1c37c62d877e21ed27a387e                                                                                                                   0.0s
 => => sha256:87703314048c40236c6d674424159ee862e2b96ce1c37c62d877e21ed27a387e 1.64kB / 1.64kB                                                                                                                                         0.0s
 => => sha256:a9c28c813336ece5bb98b36af5b66209ed777a394f4f856c6e62267790883820 528B / 528B                                                                                                                                             0.0s
 => => sha256:13621d1b12d4d0d4ad6699d220cd46f26764a9daf26a95f792ab2ce0715df366 1.47kB / 1.47kB                                                                                                                                         0.0s
 => [stage-1 2/8] RUN apk add -U --no-cache     ca-certificates     bash     curl     tzdata     libc6-compat     openssl                                                                                                              1.1s
 => [stage-1 3/8] COPY --from=BASE /go/bin/ginkgo /usr/local/bin/                                                                                                                                                                      0.0s
 => [stage-1 4/8] COPY --from=BASE /usr/local/bin/helm /usr/local/bin/                                                                                                                                                                 0.1s
 => [stage-1 5/8] COPY --from=BASE /usr/local/bin/kubectl /usr/local/bin/                                                                                                                                                              0.1s
 => [stage-1 6/8] COPY --from=BASE /usr/bin/cfssl /usr/local/bin/                                                                                                                                                                      0.0s
 => [stage-1 7/8] COPY --from=BASE /usr/bin/cfssljson /usr/local/bin/                                                                                                                                                                  0.0s
 => [stage-1 8/8] COPY . /                                                                                                                                                                                                             0.1s
 => exporting to image                                                                                                                                                                                                                 0.2s
 => => exporting layers                                                                                                                                                                                                                0.2s
 => => writing image sha256:a0eea395008886cc8f2ecd6956f797080f12036f1ecd5ce5604241b63b783bce                                                                                                                                           0.0s
 => => naming to docker.io/library/nginx-ingress-controller:e2e                                                                                                                                                                        0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
[dev-env] copying docker images to cluster...
Image: "nginx-ingress-controller:e2e" with ID "sha256:a0eea395008886cc8f2ecd6956f797080f12036f1ecd5ce5604241b63b783bce" not yet present on node "ingress-nginx-dev-worker2", loading...
Image: "nginx-ingress-controller:e2e" with ID "sha256:a0eea395008886cc8f2ecd6956f797080f12036f1ecd5ce5604241b63b783bce" not yet present on node "ingress-nginx-dev-worker", loading...
Image: "ingress-controller/controller:1.0.0-dev" with ID "sha256:c64ba05221ebe3e006eb78d86e2faef1a3ebf74d5023e9dc5119530e1fa11d78" not yet present on node "ingress-nginx-dev-worker2", loading...
Image: "ingress-controller/controller:1.0.0-dev" with ID "sha256:c64ba05221ebe3e006eb78d86e2faef1a3ebf74d5023e9dc5119530e1fa11d78" not yet present on node "ingress-nginx-dev-worker", loading...
[dev-env] running e2e tests...
Granting permissions to ingress-nginx e2e service account...
serviceaccount/ingress-nginx-e2e created
clusterrolebinding.rbac.authorization.k8s.io/permissive-binding created
Waiting service account...
Starting the e2e test pod
If you don't see a command prompt, try pressing enter.

------------------------------
• [SLOW TEST:18.400 seconds]
[Setting] [Security] modsecurity-snippet should add value of modsecurity-snippet setting to nginx config
/go/src/k8s.io/ingress-nginx/test/e2e/settings/modsecurity_snippet.go:30
------------------------------
• [SLOW TEST:27.453 seconds]
[Annotations] whitelist-source-range should set valid ip whitelist range
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/ipwhitelist.go:33
------------------------------
• [SLOW TEST:27.462 seconds]
[Annotations] affinity session-cookie-name should warn user when use-regex is true and session-cookie-path is not set
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/affinity.go:247
------------------------------
• [SLOW TEST:27.470 seconds]
[Annotations] auth-* should return status code 200 when no authentication is configured
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:46
------------------------------
• [SLOW TEST:28.420 seconds]
[Annotations] modsecurity owasp should enable modsecurity with snippet
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/modsecurity.go:90
------------------------------
• [SLOW TEST:31.719 seconds]
[Annotations] auth-tls-* should validate auth-tls-verify-client
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/authtls.go:153
------------------------------
• [SLOW TEST:34.471 seconds]
[Setting] server-tokens should exists Server header in the response when is enabled
/go/src/k8s.io/ingress-nginx/test/e2e/settings/server_tokens.go:50
------------------------------
• [SLOW TEST:37.040 seconds]
[Flag] ingress-class With a specific ingress-class should delete Ingress when class is removed
/go/src/k8s.io/ingress-nginx/test/e2e/settings/ingress_class.go:173
------------------------------
• [SLOW TEST:29.115 seconds]
[Setting] log-format-* Check log-format-escape-json should disable the log-format-escape-json
/go/src/k8s.io/ingress-nginx/test/e2e/settings/log-format.go:55
------------------------------
• [SLOW TEST:30.261 seconds]
[Setting] [SSL] TLS protocols, ciphers and headers) should configure HSTS policy header setting max-age parameter
/go/src/k8s.io/ingress-nginx/test/e2e/settings/tls.go:133
------------------------------
• [SLOW TEST:25.060 seconds]
[Annotations] cors-* should expose headers for cors
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/cors.go:140
------------------------------
• [SLOW TEST:34.141 seconds]
[Annotations] canary-* when canary is created should response with a 200 status from the mainline upstream when requests are made to the mainline ingress
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:47
------------------------------
S
------------------------------
• [SLOW TEST:39.138 seconds]
[Annotations] canary-* when canary is created should route requests to the correct upstream if the mainline ingress is modified
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:249
------------------------------
• [SLOW TEST:16.983 seconds]
[Service] Type ExternalName should return status 502 for service type=ExternalName with an invalid host
/go/src/k8s.io/ingress-nginx/test/e2e/servicebackend/service_externalname.go:149
------------------------------
• [SLOW TEST:21.590 seconds]
[Default Backend] custom service uses custom default backend that returns 200 as status code
/go/src/k8s.io/ingress-nginx/test/e2e/defaultbackend/custom_default_backend.go:36
------------------------------
• [SLOW TEST:36.556 seconds]
[Setting] use-forwarded-headers should trust X-Forwarded headers when setting is true
/go/src/k8s.io/ingress-nginx/test/e2e/settings/forwarded_headers.go:40
------------------------------
• [SLOW TEST:21.951 seconds]
[Annotations] connection-proxy-header set connection header to keep-alive
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/connection.go:36
------------------------------
• [SLOW TEST:66.230 seconds]
[Setting] [Security] global-auth-url when global external authentication is configured should still return status code 200 after auth backend is deleted using cache
/go/src/k8s.io/ingress-nginx/test/e2e/settings/global_external_auth.go:153
------------------------------
• [SLOW TEST:21.949 seconds]
[Annotations] cors-* should set cors methods to only allow POST, GET
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/cors.go:60
------------------------------
• [SLOW TEST:30.971 seconds]
[Annotations] canary-* when canaried by header with value and cookie should route requests to the correct upstream
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:593
------------------------------
• [SLOW TEST:50.229 seconds]
[Setting] [Security] global-auth-url when global external authentication is configured should return status code 200 when request whitelisted (via ingress annotation) service and 401 when request protected service
/go/src/k8s.io/ingress-nginx/test/e2e/settings/global_external_auth.go:124
------------------------------
• [SLOW TEST:20.953 seconds]
[Default Backend] change default settings should apply the annotation to the default backend
/go/src/k8s.io/ingress-nginx/test/e2e/defaultbackend/with_hosts.go:38
------------------------------
• [SLOW TEST:21.070 seconds]
[Annotations] server-alias should return status code 200 for host 'foo' and 'bar'
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/alias.go:62
------------------------------
• [SLOW TEST:24.989 seconds]
[Annotations] auth-* when external authentication is configured should return status code 200 when signed in
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:413
------------------------------
S
------------------------------
• [SLOW TEST:12.935 seconds]
[Default Backend] SSL should return a self generated SSL certificate
/go/src/k8s.io/ingress-nginx/test/e2e/defaultbackend/ssl.go:29
------------------------------
• [SLOW TEST:34.166 seconds]
[SSL] [Flag] default-ssl-certificate uses default ssl certificate for catch-all ingress
/go/src/k8s.io/ingress-nginx/test/e2e/settings/default_ssl_certificate.go:64
------------------------------
• [SLOW TEST:20.972 seconds]
[Annotations] backend-protocol should set backend protocol to grpc:// and use grpc_pass
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/backendprotocol.go:49
------------------------------
• [SLOW TEST:22.066 seconds]
[Annotations] cors-* should allow origin for cors
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/cors.go:108
------------------------------
• [SLOW TEST:19.925 seconds]
[Annotations] backend-protocol - FastCGI should add fastcgi_param in the configuration file
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/fastcgi.go:72
------------------------------
• [SLOW TEST:78.955 seconds]
[Shutdown] Grace period shutdown /healthz should return status code 500 during shutdown grace period
/go/src/k8s.io/ingress-nginx/test/e2e/gracefulshutdown/grace_period.go:35
------------------------------
• [SLOW TEST:28.081 seconds]
[Setting] proxy-read-timeout should set valid proxy read timeouts using configmap values
/go/src/k8s.io/ingress-nginx/test/e2e/settings/proxy_read_timeout.go:36
------------------------------
• [SLOW TEST:17.934 seconds]
Debug CLI should produce valid JSON for /dbg general
/go/src/k8s.io/ingress-nginx/test/e2e/dbg/main.go:85
------------------------------
• [SLOW TEST:20.943 seconds]
[Annotations] client-body-buffer-size should set client_body_buffer_size to 1K
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/clientbodybuffersize.go:51
------------------------------
• [SLOW TEST:21.227 seconds]
[Annotations] proxy-ssl-* should set valid proxy-ssl-secret, proxy-ssl-verify to on, proxy-ssl-verify-depth to 2, and proxy-ssl-server-name to on
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxyssl.go:64
------------------------------
• [SLOW TEST:22.240 seconds]
[Annotations] proxy-ssl-* should set valid proxy-ssl-secret
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxyssl.go:37
------------------------------
• [SLOW TEST:42.060 seconds]
[Lua] dynamic certificates given an ingress with TLS correctly configured falls back to using default certificate when secret gets deleted without reloading
/go/src/k8s.io/ingress-nginx/test/e2e/lua/dynamic_certificates.go:185
------------------------------
• [SLOW TEST:38.240 seconds]
[Flag] custom HTTP and HTTPS ports with a plain HTTP ingress should set X-Forwarded-Port headers accordingly when listening on a non-default HTTP port
/go/src/k8s.io/ingress-nginx/test/e2e/settings/listen_nondefault_ports.go:48
------------------------------
• [SLOW TEST:31.026 seconds]
[Annotations] rewrite-target use-regex enable-rewrite-log should use ~* location modifier if regex annotation is present
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/rewrite.go:111
------------------------------
• [SLOW TEST:22.017 seconds]
Debug CLI should list the backend servers
/go/src/k8s.io/ingress-nginx/test/e2e/dbg/main.go:37
------------------------------
• [SLOW TEST:27.952 seconds]
[Setting] server-tokens should not exists Server header in the response
/go/src/k8s.io/ingress-nginx/test/e2e/settings/server_tokens.go:38
------------------------------
• [SLOW TEST:27.973 seconds]
[Setting] keep-alive keep-alive-requests Check the keep alive should set keepalive_timeout
/go/src/k8s.io/ingress-nginx/test/e2e/settings/keep-alive.go:40
------------------------------
• [SLOW TEST:21.974 seconds]
[Annotations] app-root should redirect to /foo
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/approot.go:35
------------------------------
• [SLOW TEST:38.072 seconds]
[Annotations] auth-* when external authentication is configured keeps processing new ingresses even if one of the existing ingresses is misconfigured
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:433
------------------------------
• [SLOW TEST:28.999 seconds]
[Annotations] modsecurity owasp should enable modsecurity without using 'modsecurity on;'
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/modsecurity.go:109
------------------------------
• [SLOW TEST:34.961 seconds]
[Setting] use-forwarded-headers should not trust X-Forwarded headers when setting is false
/go/src/k8s.io/ingress-nginx/test/e2e/settings/forwarded_headers.go:92
------------------------------
• [SLOW TEST:20.980 seconds]
[Annotations] auth-* should set cache_key when external auth cache is configured
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:294
------------------------------
• [SLOW TEST:27.958 seconds]
[Annotations] rewrite-target use-regex enable-rewrite-log should fail to use longest match for documented warning
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/rewrite.go:158
------------------------------
• [SLOW TEST:28.988 seconds]
[Annotations] auth-* when external authentication is configured with a custom redirect param should redirect to signin url when not signed in
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:499
------------------------------
• [SLOW TEST:27.963 seconds]
Configure OpenTracing should exists opentracing directive when is enabled
/go/src/k8s.io/ingress-nginx/test/e2e/settings/opentracing.go:70
------------------------------
• [SLOW TEST:28.022 seconds]
[Setting] log-format-* Check log-format-upstream with log-format-escape-json log-format-escape-json disabled
/go/src/k8s.io/ingress-nginx/test/e2e/settings/log-format.go:89
------------------------------
• [SLOW TEST:38.059 seconds]
[Flag] disable-catch-all should ignore catch all Ingress with backend
/go/src/k8s.io/ingress-nginx/test/e2e/settings/disable_catch_all.go:50
------------------------------
• [SLOW TEST:30.044 seconds]
[Setting] hash size Check the variable hash size should set variables-hash-max-size
/go/src/k8s.io/ingress-nginx/test/e2e/settings/hash-size.go:88
------------------------------
• [SLOW TEST:31.022 seconds]
[Setting] settings-global-rate-limit generates correct NGINX configuration
/go/src/k8s.io/ingress-nginx/test/e2e/settings/globalratelimit.go:38
------------------------------
• [SLOW TEST:21.901 seconds]
[Annotations] canary-* when canary is created should return 404 status for requests to the canary if no matching ingress is found
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:79
------------------------------
• [SLOW TEST:21.965 seconds]
[Annotations] backend-protocol should set backend protocol to https:// and use proxy_pass
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/backendprotocol.go:34
------------------------------
• [SLOW TEST:22.980 seconds]
[Annotations] server-alias should return status code 200 for host 'foo' and 404 for 'bar'
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/alias.go:36
------------------------------
• [SLOW TEST:26.172 seconds]
[Annotations] satisfy should configure satisfy directive correctly
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/satisfy.go:42
------------------------------
• [SLOW TEST:38.085 seconds]
[Flag] ingress-class With a specific ingress-class should ignore Ingress with no class
/go/src/k8s.io/ingress-nginx/test/e2e/settings/ingress_class.go:139
------------------------------
• [SLOW TEST:39.058 seconds]
[Ingress] [PathType] exact should choose exact location for /exact
/go/src/k8s.io/ingress-nginx/test/e2e/ingress/pathtype_exact.go:37
------------------------------
• [SLOW TEST:17.964 seconds]
[Setting] enable-multi-accept should be disabled when set to false
/go/src/k8s.io/ingress-nginx/test/e2e/settings/multi_accept.go:49
------------------------------
S
------------------------------
• [SLOW TEST:25.941 seconds]
[Setting] log-format-* Check log-format-escape-json should enable the log-format-escape-json
/go/src/k8s.io/ingress-nginx/test/e2e/settings/log-format.go:47
------------------------------
• [SLOW TEST:16.957 seconds]
global-options should have worker_rlimit_nofile option and be independent on amount of worker processes
/go/src/k8s.io/ingress-nginx/test/e2e/settings/global_options.go:38
------------------------------
• [SLOW TEST:20.473 seconds]
[Security] Pod Security Policies with volumes should be running with a Pod Security Policy
/go/src/k8s.io/ingress-nginx/test/e2e/settings/pod_security_policy_volumes.go:39
------------------------------
• [SLOW TEST:28.039 seconds]
[Setting] [Security] no-auth-locations should return status code 401 when accessing '/' unauthentication
/go/src/k8s.io/ingress-nginx/test/e2e/settings/no_auth_locations.go:54
------------------------------
• [SLOW TEST:20.950 seconds]
[Annotations] modsecurity owasp should enable modsecurity with snippet and block requests
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/modsecurity.go:152
------------------------------
• [SLOW TEST:20.971 seconds]
[Annotations] auth-* should return status code 401 when authentication is configured with invalid content and Authorization header is sent
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:199
------------------------------
• [SLOW TEST:23.008 seconds]
[Annotations] configuration-snippet set snippet "more_set_headers "Request-Id: $req_id";" in all locations"
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/snippet.go:34
------------------------------
• [SLOW TEST:20.952 seconds]
[Annotations] auth-* should set snippet "proxy_set_header My-Custom-Header 42;" when external auth is configured
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:238
------------------------------
• [SLOW TEST:30.045 seconds]
[Setting] log-format-* Check log-format-upstream with log-format-escape-json log-format-escape-json enabled
/go/src/k8s.io/ingress-nginx/test/e2e/settings/log-format.go:66
------------------------------
• [SLOW TEST:29.046 seconds]
[Setting] [Security] no-auth-locations should return status code 200 when accessing '/'  authentication
/go/src/k8s.io/ingress-nginx/test/e2e/settings/no_auth_locations.go:68
------------------------------
• [SLOW TEST:20.955 seconds]
[Annotations] x-forwarded-prefix should set the X-Forwarded-Prefix to the annotation value
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/xforwardedprefix.go:35
------------------------------
• [SLOW TEST:27.967 seconds]
Configure OpenTracing should exists opentracing_location_operation_name directive when is configured
/go/src/k8s.io/ingress-nginx/test/e2e/settings/opentracing.go:129
------------------------------
• [SLOW TEST:31.971 seconds]
[Annotations] canary-* when canary is created should route requests to the correct upstream if mainline ingress is created after the canary ingress
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:205
------------------------------
• [SLOW TEST:19.940 seconds]
[Annotations] enable-access-log enable-rewrite-log set rewrite_log on
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/log.go:49
------------------------------
• [SLOW TEST:47.431 seconds]
[Lua] dynamic certificates given an ingress with TLS correctly configured picks up the updated certificate without reloading
/go/src/k8s.io/ingress-nginx/test/e2e/lua/dynamic_certificates.go:149
------------------------------
• [SLOW TEST:48.625 seconds]
[Setting] OCSP should enable OCSP and contain stapling information in the connection
/go/src/k8s.io/ingress-nginx/test/e2e/settings/ocsp/ocsp.go:49
------------------------------
• [SLOW TEST:18.993 seconds]
Dynamic $proxy_host should exist a proxy_host using the upstream-vhost annotation value
/go/src/k8s.io/ingress-nginx/test/e2e/settings/proxy_host.go:57
------------------------------
• [SLOW TEST:99.816 seconds]
[Annotations] auth-* when external authentication with caching is configured should deny login for different location on same server
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:604
------------------------------
• [SLOW TEST:21.017 seconds]
[Annotations] cors-* should set cors max-age
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/cors.go:76
------------------------------
• [SLOW TEST:69.385 seconds]
[Annotations] auth-* when external authentication with caching is configured should return status code 200 when signed in after auth backend is deleted
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:584
------------------------------
• [SLOW TEST:27.177 seconds]
[Lua] dynamic configuration configures balancer Lua middleware correctly
/go/src/k8s.io/ingress-nginx/test/e2e/lua/dynamic_configuration.go:50
------------------------------
• [SLOW TEST:22.032 seconds]
[Annotations] from-to-www-redirect should redirect from www HTTP to HTTP
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/fromtowwwredirect.go:38
------------------------------
• [SLOW TEST:16.941 seconds]
[Service] backend status code 503 should return 503 when all backend service endpoints are unavailable
/go/src/k8s.io/ingress-nginx/test/e2e/servicebackend/service_backend.go:54
------------------------------
• [SLOW TEST:27.918 seconds]
Configure OpenTracing should enable opentracing using datadog
/go/src/k8s.io/ingress-nginx/test/e2e/settings/opentracing.go:225
------------------------------
• [SLOW TEST:20.933 seconds]
[Annotations] proxy-* should turn on proxy-buffering
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxy.go:157
------------------------------
• [SLOW TEST:20.956 seconds]
[Annotations] enable-access-log enable-rewrite-log set access_log off
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/log.go:34
------------------------------
I0704 13:23:06.373985      43 request.go:668] Waited for 1.0000374s due to client-side throttling, not priority and fairness, request: GET:https://10.96.0.1:443/api/v1/namespaces/e2e-tests-limit-connections-1625404960332773800-9bvmc/services/nginx-ingress-controller


• [SLOW TEST:35.051 seconds]
[Annotations] Annotation - limit-connections should limit-connections
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/limitconnections.go:38
------------------------------
• [SLOW TEST:29.105 seconds]
[SSL] secret update should not appear references to secret updates not used in ingress rules
/go/src/k8s.io/ingress-nginx/test/e2e/ssl/secret_update.go:40
------------------------------
• [SLOW TEST:21.941 seconds]
[Annotations] proxy-* should change the default proxy HTTP version
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxy.go:233
------------------------------
• [SLOW TEST:42.034 seconds]
[Annotations] influxdb-* when influxdb is enabled should send the request metric to the influxdb server
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/influxdb.go:48
------------------------------
• [SLOW TEST:27.949 seconds]
[Setting] hash size Check proxy header hash size should set proxy-headers-hash-bucket-size
/go/src/k8s.io/ingress-nginx/test/e2e/settings/hash-size.go:60
------------------------------
• [SLOW TEST:41.134 seconds]
[Annotations] upstream-hash-by-* should connect to the same subset of pods
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/upstreamhashby.go:92
------------------------------
• [SLOW TEST:32.021 seconds]
[Annotations] canary-* when canaried by header with value and pattern should routes to mainline upstream when the given Regex causes error
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:555
------------------------------
• [SLOW TEST:31.991 seconds]
[Annotations] canary-* Single canary Ingress should not use canary with domain as a server
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:771
------------------------------
• [SLOW TEST:17.165 seconds]
[Service] Type ExternalName should return 200 for service type=ExternalName without a port defined
/go/src/k8s.io/ingress-nginx/test/e2e/servicebackend/service_externalname.go:74
------------------------------
• [SLOW TEST:28.027 seconds]
[Setting] proxy-send-timeout should not set invalid proxy send timeouts using configmap values
/go/src/k8s.io/ingress-nginx/test/e2e/settings/proxy_send_timeout.go:52
------------------------------
S
------------------------------
• [SLOW TEST:19.930 seconds]
[Annotations] backend-protocol - FastCGI should add fastcgi_index in the configuration file
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/fastcgi.go:55
------------------------------
S
------------------------------
• [SLOW TEST:31.971 seconds]
[Annotations] canary-* when canary is created should route requests to the correct upstream if mainline ingress is created before the canary ingress
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:160
------------------------------
• [SLOW TEST:28.045 seconds]
Configure OpenTracing should enable opentracing using zipkin
/go/src/k8s.io/ingress-nginx/test/e2e/settings/opentracing.go:144
------------------------------
• [SLOW TEST:101.428 seconds]
[Annotations] auth-* when external authentication with caching is configured should deny login for different servers
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:632
------------------------------
• [SLOW TEST:23.491 seconds]
[Service] Type ExternalName should update the external name after a service update
/go/src/k8s.io/ingress-nginx/test/e2e/servicebackend/service_externalname.go:261
------------------------------
• [SLOW TEST:41.364 seconds]
[Annotations] proxy-ssl-* proxy-ssl-location-only flag should change the nginx config server part
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxyssl.go:150
------------------------------
• [SLOW TEST:20.953 seconds]
[Annotations] cors-* should enable cors
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/cors.go:35
------------------------------
• [SLOW TEST:109.975 seconds]
[Setting] use-proxy-protocol should respect port passed by the PROXY Protocol
/go/src/k8s.io/ingress-nginx/test/e2e/settings/proxy_protocol.go:46
------------------------------
• [SLOW TEST:20.942 seconds]
[Annotations] upstream-vhost set host to upstreamvhost.bar.com
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/upstreamvhost.go:34
------------------------------
• [SLOW TEST:20.935 seconds]
[Setting] log-format-* Check log-format-escape-json should disable the log-format-escape-json by default
/go/src/k8s.io/ingress-nginx/test/e2e/settings/log-format.go:40
------------------------------
• [SLOW TEST:17.138 seconds]
[Service] Type ExternalName should return 200 for service type=ExternalName with a port defined
/go/src/k8s.io/ingress-nginx/test/e2e/servicebackend/service_externalname.go:108
------------------------------
• [SLOW TEST:31.099 seconds]
[Lua] dynamic configuration when only backends change handles endpoints only changes
/go/src/k8s.io/ingress-nginx/test/e2e/lua/dynamic_configuration.go:62
------------------------------
• [SLOW TEST:32.066 seconds]
[Setting] [Load Balancer] EWMA does not fail requests
/go/src/k8s.io/ingress-nginx/test/e2e/loadbalance/ewma.go:42
------------------------------
• [SLOW TEST:26.030 seconds]
[Annotations] from-to-www-redirect should redirect from www HTTPS to HTTPS
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/fromtowwwredirect.go:64
------------------------------
• [SLOW TEST:30.025 seconds]
[Setting] [SSL] TLS protocols, ciphers and headers) should configure HSTS policy header overriding what's set from the upstream
/go/src/k8s.io/ingress-nginx/test/e2e/settings/tls.go:188
------------------------------
• [SLOW TEST:16.955 seconds]
[Annotations] permanent-redirect permanent-redirect-code should respond with a standard redirect code
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/redirect.go:33
------------------------------
• [SLOW TEST:20.936 seconds]
[Annotations] proxy-* should not set proxy client-max-body-size to incorrect value
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxy.go:100
------------------------------
• [SLOW TEST:20.983 seconds]
[Annotations] client-body-buffer-size should not set client_body_buffer_size to invalid 1b
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/clientbodybuffersize.go:115
------------------------------
• [SLOW TEST:41.080 seconds]
[Setting] Configmap - limit-rate Check limit-rate config
/go/src/k8s.io/ingress-nginx/test/e2e/settings/limit_rate.go:36
------------------------------
• [SLOW TEST:32.087 seconds]
[Annotations] rewrite-target use-regex enable-rewrite-log should use correct longest path match
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/rewrite.go:66
------------------------------
• [SLOW TEST:20.917 seconds]
[Annotations] proxy-* should set proxy_redirect to off
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxy.go:36
------------------------------
• [SLOW TEST:17.017 seconds]
[Setting] enable-multi-accept should be enabled when set to true
/go/src/k8s.io/ingress-nginx/test/e2e/settings/multi_accept.go:39
------------------------------
• [SLOW TEST:55.170 seconds]
[Setting] [Security] global-auth-url when global external authentication is configured should add custom error page when global-auth-signin url is configured
/go/src/k8s.io/ingress-nginx/test/e2e/settings/global_external_auth.go:208
------------------------------
• [SLOW TEST:25.043 seconds]
[Annotations] satisfy should allow multiple auth with satisfy any
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/satisfy.go:84
------------------------------
• [SLOW TEST:16.931 seconds]
[Annotations] permanent-redirect permanent-redirect-code should respond with a custom redirect code
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/redirect.go:61
------------------------------
• [SLOW TEST:27.984 seconds]
[Annotations] server-alias should return status code 200 for hosts defined in two ingresses, different path with one alias
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/alias.go:87
------------------------------
• [SLOW TEST:20.938 seconds]
[Annotations] affinity session-cookie-name should set sticky cookie without host
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/affinity.go:305
------------------------------
• [SLOW TEST:27.937 seconds]
[Setting] [Security] block-* should block CIDRs defined in the ConfigMap
/go/src/k8s.io/ingress-nginx/test/e2e/settings/global_access_block.go:38
------------------------------
• [SLOW TEST:16.283 seconds]
[Default Backend] should return 404 sending requests when only a default backend is running
/go/src/k8s.io/ingress-nginx/test/e2e/defaultbackend/default_backend.go:32
------------------------------
• [SLOW TEST:32.993 seconds]
[Annotations] canary-* when canary is created should route requests to the correct upstream if the canary ingress is modified
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:306
------------------------------
• [SLOW TEST:28.957 seconds]
[Shutdown] ingress controller should shutdown in less than 60 secons without pending connections
/go/src/k8s.io/ingress-nginx/test/e2e/gracefulshutdown/shutdown.go:40
------------------------------
• [SLOW TEST:27.977 seconds]
[Setting] add-headers Add multiple custom headers
/go/src/k8s.io/ingress-nginx/test/e2e/settings/custom_header.go:65
------------------------------
• [SLOW TEST:9.916 seconds]
global-options should have worker_rlimit_nofile option
/go/src/k8s.io/ingress-nginx/test/e2e/settings/global_options.go:31
------------------------------
• [SLOW TEST:20.936 seconds]
[Annotations] auth-* should return status code 401 when authentication is configured and Authorization header is sent with invalid credentials
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:116
------------------------------
• [SLOW TEST:25.985 seconds]
[Annotations] auth-* when external authentication is configured should redirect to signin url when not signed in
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:422
------------------------------
• [SLOW TEST:20.949 seconds]
[Annotations] affinity session-cookie-name should set cookie with expires
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/affinity.go:192
------------------------------
• [SLOW TEST:55.186 seconds]
[Setting] [Security] global-auth-url when global external authentication is configured should add auth headers when global-auth-response-headers is configured
/go/src/k8s.io/ingress-nginx/test/e2e/settings/global_external_auth.go:221
------------------------------
• [SLOW TEST:25.954 seconds]
[Setting] keep-alive keep-alive-requests Check the upstream keep alive should set keepalive connection to upstream server
/go/src/k8s.io/ingress-nginx/test/e2e/settings/keep-alive.go:59
------------------------------
• [SLOW TEST:28.991 seconds]
[Setting] [Security] block-* should block Referers defined in the ConfigMap
/go/src/k8s.io/ingress-nginx/test/e2e/settings/global_access_block.go:88
------------------------------
• [SLOW TEST:20.944 seconds]
Dynamic $proxy_host should exist a proxy_host
/go/src/k8s.io/ingress-nginx/test/e2e/settings/proxy_host.go:36
------------------------------
• [SLOW TEST:32.944 seconds]
[Security] request smuggling should not return body content from error_page
/go/src/k8s.io/ingress-nginx/test/e2e/security/request_smuggling.go:39
------------------------------
S
------------------------------
• [SLOW TEST:19.972 seconds]
[Annotations] backend-protocol should set backend protocol to '' and use fastcgi_pass
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/backendprotocol.go:79
------------------------------
• [SLOW TEST:29.047 seconds]
plugins should exist a x-hello-world header
/go/src/k8s.io/ingress-nginx/test/e2e/settings/plugins.go:35
------------------------------
• [SLOW TEST:27.950 seconds]
Configure OpenTracing should not exists opentracing directive
/go/src/k8s.io/ingress-nginx/test/e2e/settings/opentracing.go:57
------------------------------
• [SLOW TEST:20.951 seconds]
[Annotations] default-backend when default backend annotation is enabled should use a custom default backend as upstream
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/default_backend.go:37
------------------------------
• [SLOW TEST:41.084 seconds]
[Lua] dynamic certificates given an ingress with TLS correctly configured removes HTTPS configuration when we delete TLS spec
/go/src/k8s.io/ingress-nginx/test/e2e/lua/dynamic_certificates.go:233
------------------------------
• [SLOW TEST:17.179 seconds]
[Service] Type ExternalName should return 200 for service type=ExternalName using a port name
/go/src/k8s.io/ingress-nginx/test/e2e/servicebackend/service_externalname.go:180
------------------------------
• [SLOW TEST:21.056 seconds]
[Annotations] proxy-* should build proxy next upstream
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxy.go:194
------------------------------
• [SLOW TEST:110.965 seconds]
[Setting] use-proxy-protocol should respect proto passed by the PROXY Protocol server port
/go/src/k8s.io/ingress-nginx/test/e2e/settings/proxy_protocol.go:79
------------------------------
• [SLOW TEST:28.047 seconds]
[Flag] disable-catch-all should allow Ingress with rules
/go/src/k8s.io/ingress-nginx/test/e2e/settings/disable_catch_all.go:123
------------------------------
• [SLOW TEST:19.595 seconds]
[Annotations] auth-* should return status code 200 when authentication is configured and Authorization header is sent
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:144
------------------------------
• [SLOW TEST:20.947 seconds]
[Annotations] proxy-* should set proxy client-max-body-size to 8m
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxy.go:85
------------------------------
• [SLOW TEST:22.396 seconds]
[Annotations] client-body-buffer-size should set client_body_buffer_size to 1k
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/clientbodybuffersize.go:67
------------------------------
• [SLOW TEST:21.613 seconds]
[Annotations] auth-* should not set snippet "proxy_set_header My-Custom-Header 42;" when external auth is not configured
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:256
------------------------------
• [SLOW TEST:27.986 seconds]
[Flag] disable-catch-all should ignore catch all Ingress with backend and rules
/go/src/k8s.io/ingress-nginx/test/e2e/settings/disable_catch_all.go:69
------------------------------
• [SLOW TEST:145.958 seconds]
[Setting] [Load Balancer] round-robin should evenly distribute requests with round-robin (default algorithm)
/go/src/k8s.io/ingress-nginx/test/e2e/loadbalance/round_robin.go:39
------------------------------
• [SLOW TEST:21.010 seconds]
[Annotations] affinity session-cookie-name should work with use-regex annotation and session-cookie-path
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/affinity.go:223
------------------------------
• [SLOW TEST:39.017 seconds]
Configure OpenTracing should propagate the w3c header when configured with jaeger
/go/src/k8s.io/ingress-nginx/test/e2e/settings/opentracing.go:181
------------------------------
• [SLOW TEST:28.016 seconds]
[Setting] proxy-read-timeout should not set invalid proxy read timeouts using configmap values
/go/src/k8s.io/ingress-nginx/test/e2e/settings/proxy_read_timeout.go:52
------------------------------
• [SLOW TEST:27.961 seconds]
[Setting] hash size Check proxy header hash size should set proxy-headers-hash-max-size
/go/src/k8s.io/ingress-nginx/test/e2e/settings/hash-size.go:68
------------------------------
• [SLOW TEST:125.380 seconds]
[Annotations] affinitymode Check persistent affinity mode
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/affinitymode.go:64
------------------------------
• [SLOW TEST:27.932 seconds]
[Setting] hash size Check the map hash size should set vmap-hash-bucket-size
/go/src/k8s.io/ingress-nginx/test/e2e/settings/hash-size.go:100
------------------------------
• [SLOW TEST:20.942 seconds]
[Annotations] client-body-buffer-size should set client_body_buffer_size to 1000
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/clientbodybuffersize.go:35
------------------------------
• [SLOW TEST:20.955 seconds]
[Annotations] x-forwarded-prefix should not add X-Forwarded-Prefix if the annotation value is empty
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/xforwardedprefix.go:57
------------------------------
• [SLOW TEST:18.947 seconds]
[SSL] secret update should return the fake SSL certificate if the secret is invalid
/go/src/k8s.io/ingress-nginx/test/e2e/ssl/secret_update.go:82
------------------------------
• [SLOW TEST:32.017 seconds]
[Annotations] annotation-global-rate-limit generates correct configuration
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/globalratelimit.go:38
------------------------------
• [SLOW TEST:26.943 seconds]
[Setting] keep-alive keep-alive-requests Check the upstream keep alive should set keep alive connection timeout to upstream server
/go/src/k8s.io/ingress-nginx/test/e2e/settings/keep-alive.go:68
------------------------------
• [SLOW TEST:10.923 seconds]
[Setting] enable-multi-accept should be enabled by default
/go/src/k8s.io/ingress-nginx/test/e2e/settings/multi_accept.go:31
------------------------------
• [SLOW TEST:112.074 seconds]
[Setting] use-proxy-protocol should enable PROXY Protocol for HTTPS
/go/src/k8s.io/ingress-nginx/test/e2e/settings/proxy_protocol.go:112
------------------------------
• [SLOW TEST:26.128 seconds]
[Lua] dynamic certificates picks up the certificate when we add TLS spec to existing ingress
/go/src/k8s.io/ingress-nginx/test/e2e/lua/dynamic_certificates.go:45
------------------------------
• [SLOW TEST:25.049 seconds]
[Annotations] auth-* when external authentication is configured with a custom redirect param should return status code 200 when signed in
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:490
------------------------------
• [SLOW TEST:31.311 seconds]
[Annotations] canary-* when canaried by cookie should route requests to the correct upstream
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:632
------------------------------
• [SLOW TEST:16.930 seconds]
[Setting] main-snippet should add value of main-snippet setting to nginx config
/go/src/k8s.io/ingress-nginx/test/e2e/settings/main_snippet.go:31
------------------------------
• [SLOW TEST:45.138 seconds]
[Lua] dynamic configuration when only backends change handles endpoints only changes (down scaling of replicas)
/go/src/k8s.io/ingress-nginx/test/e2e/lua/dynamic_configuration.go:87
------------------------------
• [SLOW TEST:34.274 seconds]
[Annotations] canary-* does not crash when canary ingress has multiple paths to the same non-matching backend
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:795
------------------------------
• [SLOW TEST:76.670 seconds]
[Setting] [Security] global-auth-url when global external authentication is configured should return status code 200 when request whitelisted (via no-auth-locations) service and 401 when request protected service
/go/src/k8s.io/ingress-nginx/test/e2e/settings/global_external_auth.go:100
------------------------------
• [SLOW TEST:34.950 seconds]
[Setting] enable-real-ip trusts X-Forwarded-For header only when setting is true
/go/src/k8s.io/ingress-nginx/test/e2e/settings/enable_real_ip.go:40
------------------------------
• [SLOW TEST:21.943 seconds]
[Annotations] client-body-buffer-size should set client_body_buffer_size to 1m
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/clientbodybuffersize.go:83
------------------------------
• [SLOW TEST:39.157 seconds]
[Setting] Configmap change should reload after an update in the configuration
/go/src/k8s.io/ingress-nginx/test/e2e/settings/configmap_change.go:36
------------------------------
• [SLOW TEST:25.948 seconds]
[Setting] keep-alive keep-alive-requests Check the upstream keep alive should set the request count to upstream server through one keep alive connection
/go/src/k8s.io/ingress-nginx/test/e2e/settings/keep-alive.go:77
------------------------------
S
------------------------------
• [SLOW TEST:20.939 seconds]
[Annotations] http2-push-preload enable the http2-push-preload directive
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/http2pushpreload.go:34
------------------------------
• [SLOW TEST:21.924 seconds]
[Annotations] ssl-ciphers should change ssl ciphers
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/sslciphers.go:34
------------------------------
• [SLOW TEST:20.938 seconds]
[Annotations] modsecurity owasp should enable modsecurity with transaction ID and OWASP rules
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/modsecurity.go:52
------------------------------
• [SLOW TEST:32.984 seconds]
[Annotations] canary-* when canaried by header with value and pattern should route requests to the correct upstream
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:521
------------------------------
• [SLOW TEST:20.939 seconds]
[Annotations] affinity session-cookie-name does not set the path to / on the generated cookie if there's more than one rule referring to the same backend
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/affinity.go:120
------------------------------
• [SLOW TEST:27.972 seconds]
[Flag] ingress-class Without a specific ingress-class should ignore Ingress with class
/go/src/k8s.io/ingress-nginx/test/e2e/settings/ingress_class.go:87
------------------------------
• [SLOW TEST:32.003 seconds]
[Annotations] canary-* Single canary Ingress should not use canary as a catch-all server
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:743
------------------------------
• [SLOW TEST:20.958 seconds]
[Ingress] [PathType] prefix checks should return 404 when prefix /aaa does not match request /aaaccc
/go/src/k8s.io/ingress-nginx/test/e2e/ingress/pathtype_prefix.go:35
------------------------------
• [SLOW TEST:17.938 seconds]
[Setting] ssl-ciphers Add ssl ciphers
/go/src/k8s.io/ingress-nginx/test/e2e/settings/ssl_ciphers.go:31
------------------------------
• [SLOW TEST:45.091 seconds]
[Setting] [Security] global-auth-url when global external authentication is configured should return status code 401 when request any protected service
/go/src/k8s.io/ingress-nginx/test/e2e/settings/global_external_auth.go:83
------------------------------
• [SLOW TEST:36.025 seconds]
[Annotations] auth-* when external authentication is configured with a custom redirect param keeps processing new ingresses even if one of the existing ingresses is misconfigured
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:510
------------------------------
• [SLOW TEST:43.027 seconds]
[Ingress] [PathType] mix Exact and Prefix paths should choose the correct location
/go/src/k8s.io/ingress-nginx/test/e2e/ingress/pathtype_mixed.go:39
------------------------------
• [SLOW TEST:61.199 seconds]
[Annotations] auth-* when external authentication with caching is configured should redirect to signin url when not signed in
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:661
------------------------------
• [SLOW TEST:21.943 seconds]
[Annotations] mirror-* should disable mirror-request-body
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/mirror.go:67
------------------------------
• [SLOW TEST:27.941 seconds]
[Setting] keep-alive keep-alive-requests Check the keep alive should set keepalive_requests
/go/src/k8s.io/ingress-nginx/test/e2e/settings/keep-alive.go:48
------------------------------
• [SLOW TEST:30.033 seconds]
[Setting] [SSL] TLS protocols, ciphers and headers) should configure TLS protocol enforcing TLS v1.0
/go/src/k8s.io/ingress-nginx/test/e2e/settings/tls.go:87
------------------------------
• [SLOW TEST:28.963 seconds]
[Annotations] modsecurity owasp should disable modsecurity using 'modsecurity off;'
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/modsecurity.go:131
------------------------------
• [SLOW TEST:22.230 seconds]
[Annotations] auth-tls-* should set valid auth-tls-secret
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/authtls.go:37
------------------------------
• [SLOW TEST:29.000 seconds]
[Annotations] modsecurity owasp should enable modsecurity globally and with modsecurity-snippet block requests
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/modsecurity.go:186
------------------------------
• [SLOW TEST:19.936 seconds]
[Annotations] mirror-* should set mirror-target to http://localhost/mirror
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/mirror.go:36
------------------------------
• [SLOW TEST:30.962 seconds]
[Annotations] canary-* when canaried by header with value should route requests to the correct upstream
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:415
------------------------------
• [SLOW TEST:25.958 seconds]
[Annotations] backend-protocol should set backend protocol to '' and use ajp_pass
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/backendprotocol.go:94
------------------------------
• [SLOW TEST:33.137 seconds]
[Setting] [SSL] TLS protocols, ciphers and headers) should configure TLS protocol setting cipher suite
/go/src/k8s.io/ingress-nginx/test/e2e/settings/tls.go:65
------------------------------
• [SLOW TEST:18.140 seconds]
[Service] Type ExternalName should return 200 for service type=ExternalName using FQDN with trailing dot
/go/src/k8s.io/ingress-nginx/test/e2e/servicebackend/service_externalname.go:230
------------------------------
• [SLOW TEST:23.962 seconds]
[Annotations] auth-* should return status code 200 when authentication is configured with a map and Authorization header is sent
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:171
------------------------------
• [SLOW TEST:29.955 seconds]
[Setting] [Security] no-auth-locations should return status code 200 when accessing '/noauth' unauthenticated
/go/src/k8s.io/ingress-nginx/test/e2e/settings/no_auth_locations.go:82
------------------------------
• [SLOW TEST:24.021 seconds]
[Annotations] modsecurity owasp should enable modsecurity
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/modsecurity.go:34
------------------------------
• [SLOW TEST:23.956 seconds]
[Annotations] affinity session-cookie-name should set the path to /something on the generated cookie
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/affinity.go:98
------------------------------
S
------------------------------
• [SLOW TEST:28.398 seconds]
Configure OpenTracing should not exists opentracing_operation_name directive when is empty
/go/src/k8s.io/ingress-nginx/test/e2e/settings/opentracing.go:84
------------------------------
• [SLOW TEST:24.934 seconds]
[Annotations] client-body-buffer-size should set client_body_buffer_size to 1M
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/clientbodybuffersize.go:99
------------------------------
• [SLOW TEST:25.052 seconds]
[Annotations] affinity session-cookie-name should set sticky cookie SERVERID
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/affinity.go:41
------------------------------
• [SLOW TEST:30.115 seconds]
[Annotations] limit-rate Check limit-rate annotation
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/limitrate.go:37
------------------------------
• [SLOW TEST:33.501 seconds]
[SSL] [Flag] default-ssl-certificate uses default ssl certificate for host based ingress when configured certificate does not match host
/go/src/k8s.io/ingress-nginx/test/e2e/settings/default_ssl_certificate.go:80
------------------------------
• [SLOW TEST:42.208 seconds]
[Flag] custom HTTP and HTTPS ports with a TLS enabled ingress when external authentication is configured should set the X-Forwarded-Port header to 443
/go/src/k8s.io/ingress-nginx/test/e2e/settings/listen_nondefault_ports.go:100
------------------------------
• [SLOW TEST:20.952 seconds]
[Annotations] server-snippet add valid directives to server via server snippet
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/serversnippet.go:34
------------------------------
• [SLOW TEST:19.938 seconds]
[Annotations] modsecurity owasp should disable modsecurity
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/modsecurity.go:73
------------------------------
• [SLOW TEST:30.111 seconds]
single ingress - multiple hosts should set the correct $service_name NGINX variable
/go/src/k8s.io/ingress-nginx/test/e2e/ingress/multiple_rules.go:38
------------------------------
• [SLOW TEST:27.948 seconds]
Configure OpenTracing should not exists opentracing_location_operation_name directive when is empty
/go/src/k8s.io/ingress-nginx/test/e2e/settings/opentracing.go:114
------------------------------
S
------------------------------
• [SLOW TEST:18.946 seconds]
[Annotations] proxy-* should set proxy_redirect to hello.com goodbye.com
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxy.go:68
------------------------------
• [SLOW TEST:26.979 seconds]
[Setting] [Security] block-* should block User-Agents defined in the ConfigMap
/go/src/k8s.io/ingress-nginx/test/e2e/settings/global_access_block.go:55
------------------------------
• [SLOW TEST:27.938 seconds]
[Setting] add-headers Add a custom header
/go/src/k8s.io/ingress-nginx/test/e2e/settings/custom_header.go:40
------------------------------
• [SLOW TEST:57.217 seconds]
[Setting] [Security] global-auth-url when global external authentication is configured should set request-redirect when global-auth-request-redirect is configured
/go/src/k8s.io/ingress-nginx/test/e2e/settings/global_external_auth.go:235
------------------------------
• [SLOW TEST:21.928 seconds]
[Annotations] proxy-* should set valid proxy timeouts
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxy.go:115
------------------------------
• [SLOW TEST:54.185 seconds]
[Flag] ingress-class check scenarios for IngressClass and ingress.class annotation
/go/src/k8s.io/ingress-nginx/test/e2e/settings/ingress_class.go:212
------------------------------
• [SLOW TEST:10.950 seconds]
[Setting] access-log access-log-path use the default configuration
/go/src/k8s.io/ingress-nginx/test/e2e/settings/access_log.go:31
------------------------------
• [SLOW TEST:33.012 seconds]
[Annotations] canary-* when canaried by weight should route requests to the correct upstream
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:687
------------------------------
• [SLOW TEST:25.979 seconds]
[Annotations] affinity session-cookie-name should change cookie name on ingress definition change
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/affinity.go:63
------------------------------
• [SLOW TEST:27.979 seconds]
[Setting] hash size Check the variable hash size should set variables-hash-bucket-size
/go/src/k8s.io/ingress-nginx/test/e2e/settings/hash-size.go:80
------------------------------
• [SLOW TEST:42.260 seconds]
[Flag] custom HTTP and HTTPS ports with a TLS enabled ingress should set X-Forwarded-Port header to 443
/go/src/k8s.io/ingress-nginx/test/e2e/settings/listen_nondefault_ports.go:70
------------------------------
• [SLOW TEST:20.945 seconds]
[Ingress] definition without host should set ingress details variables for ingresses without a host
/go/src/k8s.io/ingress-nginx/test/e2e/ingress/without_host.go:34
------------------------------
• [SLOW TEST:22.959 seconds]
[Annotations] auth-* retains cookie set by external authentication server
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:316
------------------------------
• [SLOW TEST:21.977 seconds]
[Annotations] cors-* should allow headers for cors
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/cors.go:124
------------------------------
• [SLOW TEST:19.974 seconds]
[Setting] [SSL] TLS protocols, ciphers and headers) ports or X-Forwarded-Host check during HTTP tp HTTPS redirection should not use ports during the HTTP to HTTPS redirection
/go/src/k8s.io/ingress-nginx/test/e2e/settings/tls.go:209
------------------------------
• [SLOW TEST:19.097 seconds]
[Annotations] backend-protocol - FastCGI should return OK for service with backend protocol FastCGI
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/fastcgi.go:105
------------------------------
• [SLOW TEST:27.937 seconds]
[Setting] proxy-connect-timeout should set valid proxy timeouts using configmap values
/go/src/k8s.io/ingress-nginx/test/e2e/settings/proxy_connect_timeout.go:36
------------------------------
• [SLOW TEST:27.965 seconds]
[Setting] proxy-connect-timeout should not set invalid proxy timeouts using configmap values
/go/src/k8s.io/ingress-nginx/test/e2e/settings/proxy_connect_timeout.go:52
------------------------------
• [SLOW TEST:20.962 seconds]
[SSL] redirect to HTTPS should redirect from HTTP to HTTPS when secret is missing
/go/src/k8s.io/ingress-nginx/test/e2e/ssl/http_redirect.go:36
------------------------------
• [SLOW TEST:27.137 seconds]
[Lua] dynamic configuration when only backends change handles an annotation change
/go/src/k8s.io/ingress-nginx/test/e2e/lua/dynamic_configuration.go:168
------------------------------
• [SLOW TEST:11.166 seconds]
[TCP] tcp-services should expose an ExternalName TCP service
/go/src/k8s.io/ingress-nginx/test/e2e/tcpudp/tcp.go:98
------------------------------
• [SLOW TEST:28.123 seconds]
[Setting] hash size Check server names hash size should set server_names_hash_max_size
/go/src/k8s.io/ingress-nginx/test/e2e/settings/hash-size.go:48
------------------------------
• [SLOW TEST:18.994 seconds]
[Annotations] auth-* should return status code 401 when authentication is configured but Authorization header is not configured
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:89
------------------------------
• [SLOW TEST:28.977 seconds]
Configure OpenTracing should exists opentracing_operation_name directive when is configured
/go/src/k8s.io/ingress-nginx/test/e2e/settings/opentracing.go:99
------------------------------
• [SLOW TEST:23.975 seconds]
[Annotations] affinitymode Balanced affinity mode should balance
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/affinitymode.go:34
------------------------------
• [SLOW TEST:27.086 seconds]
[Setting] Add no tls redirect locations Check no tls redirect locations config
/go/src/k8s.io/ingress-nginx/test/e2e/settings/no_tls_redirect_locations.go:31
------------------------------
• [SLOW TEST:47.255 seconds]
[Lua] dynamic configuration when only backends change handles endpoints only changes consistently (down scaling of replicas vs. empty service)
/go/src/k8s.io/ingress-nginx/test/e2e/lua/dynamic_configuration.go:125
------------------------------
• [SLOW TEST:21.954 seconds]
[Ingress] definition without host should set ingress details variables for ingresses with host without IngressRuleValue, only Backend
/go/src/k8s.io/ingress-nginx/test/e2e/ingress/without_host.go:55
------------------------------
• [SLOW TEST:22.064 seconds]
[Annotations] proxy-* should setup proxy cookies
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxy.go:215
------------------------------
• [SLOW TEST:21.995 seconds]
[Annotations] backend-protocol should set backend protocol to grpcs:// and use grpc_pass
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/backendprotocol.go:64
------------------------------
• [SLOW TEST:28.084 seconds]
Configure OpenTracing should enable opentracing using jaeger
/go/src/k8s.io/ingress-nginx/test/e2e/settings/opentracing.go:156
------------------------------
• [SLOW TEST:23.006 seconds]
[Annotations] backend-protocol - FastCGI should use fastcgi_pass in the configuration file
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/fastcgi.go:38
------------------------------
• [SLOW TEST:25.963 seconds]
[Annotations] mirror-* should set mirror-target to https://test.env.com/$request_uri
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/mirror.go:51
------------------------------
• [SLOW TEST:31.132 seconds]
[Lua] dynamic certificates picks up the previously missing secret for a given ingress without reloading
/go/src/k8s.io/ingress-nginx/test/e2e/lua/dynamic_certificates.go:70
------------------------------
• [SLOW TEST:21.456 seconds]
[Annotations] proxy-ssl-* should set valid proxy-ssl-secret, proxy-ssl-ciphers to HIGH:!AES
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxyssl.go:94
------------------------------
• [SLOW TEST:39.983 seconds]
[Setting] enable-real-ip should not trust X-Forwarded-For header when setting is false
/go/src/k8s.io/ingress-nginx/test/e2e/settings/enable_real_ip.go:78
------------------------------
• [SLOW TEST:16.924 seconds]
[Setting] access-log stream-access-log-path use the specified configuration
/go/src/k8s.io/ingress-nginx/test/e2e/settings/access_log.go:63
------------------------------
• [SLOW TEST:48.170 seconds]
[Annotations] upstream-hash-by-* should connect to the same pod
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/upstreamhashby.go:83
------------------------------
• [SLOW TEST:48.147 seconds]
[Lua] dynamic certificates given an ingress with TLS correctly configured picks up a non-certificate only change
/go/src/k8s.io/ingress-nginx/test/e2e/lua/dynamic_certificates.go:218
------------------------------
• [SLOW TEST:30.149 seconds]
[Setting] [SSL] TLS protocols, ciphers and headers) should configure HSTS policy header setting includeSubDomains parameter
/go/src/k8s.io/ingress-nginx/test/e2e/settings/tls.go:149
------------------------------
• [SLOW TEST:21.942 seconds]
[Annotations] rewrite-target use-regex enable-rewrite-log should allow for custom rewrite parameters
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/rewrite.go:190
------------------------------
• [SLOW TEST:41.174 seconds]
[Annotations] custom-http-errors configures Nginx correctly
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/customhttperrors.go:41
------------------------------
• [SLOW TEST:29.039 seconds]
[Setting] hash size Check server names hash size should set server_names_hash_bucket_size
/go/src/k8s.io/ingress-nginx/test/e2e/settings/hash-size.go:40
------------------------------
• [SLOW TEST:33.064 seconds]
[Annotations] canary-* when canaried by header with no value should route requests to the correct upstream
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:361
------------------------------
• [SLOW TEST:53.176 seconds]
[Setting] [Security] global-auth-url when global external authentication is configured should set snippet when global external auth is configured
/go/src/k8s.io/ingress-nginx/test/e2e/settings/global_external_auth.go:248
------------------------------
• [SLOW TEST:26.921 seconds]
[Setting] [SSL] TLS protocols, ciphers and headers) ports or X-Forwarded-Host check during HTTP tp HTTPS redirection should not use ports or X-Forwarded-Host during the HTTP to HTTPS redirection
/go/src/k8s.io/ingress-nginx/test/e2e/settings/tls.go:227
------------------------------
• [SLOW TEST:28.031 seconds]
[Setting] [Load Balancer] load-balance should apply the configmap load-balance setting
/go/src/k8s.io/ingress-nginx/test/e2e/loadbalance/configmap.go:35
------------------------------
• [SLOW TEST:27.005 seconds]
[Annotations] disable-access-log disable-http-access-log disable-stream-access-log disable-http-access-log set access_log off
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/disableaccesslog.go:46
------------------------------
• [SLOW TEST:25.880 seconds]
Configure OpenTracing should enable opentracing using jaeger with sampler host
/go/src/k8s.io/ingress-nginx/test/e2e/settings/opentracing.go:168
------------------------------
• [SLOW TEST:17.870 seconds]
[Annotations] force-ssl-redirect should redirect to https
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/forcesslredirect.go:34
------------------------------
• [SLOW TEST:22.140 seconds]
[Annotations] proxy-ssl-* should set valid proxy-ssl-secret, proxy-ssl-protocols
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxyssl.go:122
------------------------------
• [SLOW TEST:27.928 seconds]
[Annotations] affinity session-cookie-name should not set affinity across all server locations when using separate ingresses
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/affinity.go:273
------------------------------
• [SLOW TEST:24.414 seconds]
[Annotations] auth-tls-* should set valid auth-tls-secret, pass certificate to upstream, and error page
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/authtls.go:103
------------------------------
• [SLOW TEST:25.971 seconds]
[Annotations] disable-access-log disable-http-access-log disable-stream-access-log disable-access-log set access_log off
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/disableaccesslog.go:34
------------------------------
• [SLOW TEST:19.996 seconds]
[Setting] [Lua] lua-shared-dicts configures lua shared dicts
/go/src/k8s.io/ingress-nginx/test/e2e/settings/lua_shared_dicts.go:29
------------------------------
• [SLOW TEST:20.939 seconds]
[Annotations] cors-* should disable cors allow credentials
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/cors.go:92
------------------------------
• [SLOW TEST:16.946 seconds]
[Service] backend status code 503 should return 503 when backend service does not exist
/go/src/k8s.io/ingress-nginx/test/e2e/servicebackend/service_backend.go:36
------------------------------
• [SLOW TEST:118.018 seconds]
[Status] status update should update status field after client-go reconnection
/go/src/k8s.io/ingress-nginx/test/e2e/status/update.go:43
------------------------------
• [SLOW TEST:17.926 seconds]
[Setting] access-log access-log-path use the specified configuration
/go/src/k8s.io/ingress-nginx/test/e2e/settings/access_log.go:39
------------------------------
S
------------------------------
• [SLOW TEST:18.972 seconds]
[TCP] tcp-services should expose a TCP service
/go/src/k8s.io/ingress-nginx/test/e2e/tcpudp/tcp.go:40
------------------------------
• [SLOW TEST:20.956 seconds]
[Setting] reuse-port reuse port should be enabled by default
/go/src/k8s.io/ingress-nginx/test/e2e/settings/reuse-port.go:38
------------------------------
• [SLOW TEST:53.190 seconds]
[Setting] [Security] global-auth-url when global external authentication is configured should proxy_method method when global-auth-method is configured
/go/src/k8s.io/ingress-nginx/test/e2e/settings/global_external_auth.go:195
------------------------------
• [SLOW TEST:22.336 seconds]
[Annotations] auth-tls-* should set valid auth-tls-secret, sslVerify to off, and sslVerifyDepth to 2
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/authtls.go:73
------------------------------
• [SLOW TEST:24.000 seconds]
[Annotations] affinity session-cookie-name should work with server-alias annotation
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/affinity.go:325
------------------------------
• [SLOW TEST:23.490 seconds]
[Security] Pod Security Policies should be running with a Pod Security Policy
/go/src/k8s.io/ingress-nginx/test/e2e/settings/pod_security_policy.go:43
------------------------------
• [SLOW TEST:17.025 seconds]
[Setting] access-log http-access-log-path use the specified configuration
/go/src/k8s.io/ingress-nginx/test/e2e/settings/access_log.go:51
------------------------------
• [SLOW TEST:21.034 seconds]
[Annotations] proxy-* should not set invalid proxy timeouts
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxy.go:136
------------------------------
• [SLOW TEST:21.952 seconds]
[Annotations] auth-* should set "proxy_set_header 'My-Custom-Header' '42';" when auth-headers are set
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:273
------------------------------
• [SLOW TEST:24.953 seconds]
[Annotations] backend-protocol - GRPC should return OK for service with backend protocol GRPC
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/grpc.go:66
------------------------------
• [SLOW TEST:32.969 seconds]
[Annotations] canary-* when canaried by header with value and pattern should route requests to the correct upstream
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/canary.go:479
------------------------------
• [SLOW TEST:21.080 seconds]
Debug CLI should get information for a specific backend server
/go/src/k8s.io/ingress-nginx/test/e2e/dbg/main.go:56
------------------------------
• [SLOW TEST:16.937 seconds]
[Setting] access-log http-access-log-path & stream-access-log-path use the specified configuration
/go/src/k8s.io/ingress-nginx/test/e2e/settings/access_log.go:75
------------------------------
• [SLOW TEST:20.932 seconds]
[Annotations] proxy-* should set proxy_redirect to default
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxy.go:52
------------------------------
• [SLOW TEST:20.953 seconds]
[Annotations] rewrite-target use-regex enable-rewrite-log should write rewrite logs
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/rewrite.go:37
------------------------------
• [SLOW TEST:20.953 seconds]
[Annotations] backend-protocol - GRPC should return OK for service with backend protocol GRPCS
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/grpc.go:124
------------------------------
• [SLOW TEST:20.968 seconds]
[Annotations] auth-* should return status code 503 when authentication is configured with an invalid secret
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/auth.go:65
------------------------------
• [SLOW TEST:21.975 seconds]
[Service] Type ExternalName works with external name set to incomplete fqdn
/go/src/k8s.io/ingress-nginx/test/e2e/servicebackend/service_externalname.go:41
------------------------------
• [SLOW TEST:36.083 seconds]
[Lua] dynamic certificates given an ingress with TLS correctly configured supports requests with domain with trailing dot
/go/src/k8s.io/ingress-nginx/test/e2e/lua/dynamic_certificates.go:145
------------------------------
• [SLOW TEST:23.999 seconds]
[Annotations] backend-protocol - GRPC should use grpc_pass in the configuration file
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/grpc.go:41
------------------------------
• [SLOW TEST:21.956 seconds]
[Annotations] proxy-* should turn off proxy-request-buffering
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/proxy.go:179
------------------------------
• [SLOW TEST:29.006 seconds]
[Setting] reuse-port reuse port should be disabled
/go/src/k8s.io/ingress-nginx/test/e2e/settings/reuse-port.go:44
------------------------------
• [SLOW TEST:27.971 seconds]
[Setting] proxy-send-timeout should set valid proxy send timeouts using configmap values
/go/src/k8s.io/ingress-nginx/test/e2e/settings/proxy_send_timeout.go:36
------------------------------
• [SLOW TEST:18.949 seconds]
[Annotations] preserve-trailing-slash should allow preservation of trailing slashes
/go/src/k8s.io/ingress-nginx/test/e2e/annotations/preservetrailingslash.go:34
------------------------------
• [SLOW TEST:34.082 seconds]
[Flag] disable-catch-all should delete Ingress updated to catch-all
/go/src/k8s.io/ingress-nginx/test/e2e/settings/disable_catch_all.go:81
------------------------------
• [SLOW TEST:28.940 seconds]
[Setting] reuse-port reuse port should be enabled
/go/src/k8s.io/ingress-nginx/test/e2e/settings/reuse-port.go:52
------------------------------
• [SLOW TEST:27.956 seconds]
[Setting] proxy-next-upstream should build proxy next upstream using configmap values
/go/src/k8s.io/ingress-nginx/test/e2e/settings/proxy_next_upstream.go:36
------------------------------
• [SLOW TEST:32.989 seconds]
[Setting] [SSL] TLS protocols, ciphers and headers) should configure HSTS policy header setting preload parameter
/go/src/k8s.io/ingress-nginx/test/e2e/settings/tls.go:168
------------------------------
• [SLOW TEST:630.886 seconds]
[Setting] use-proxy-protocol should enable PROXY Protocol for TCP
/go/src/k8s.io/ingress-nginx/test/e2e/settings/proxy_protocol.go:155
------------------------------
 SUCCESS! 29m30.5410896s

Ginkgo ran 1 suite in 29m30.5990507s
Test Suite Passed
Running e2e test suite with tests that require serial execution...
Jul  4 13:46:21.354: INFO: Starting e2e run "980651f9-50cc-4769-b5e7-3fd2366480e6" on Ginkgo node 1
[1625406381] nginx-ingress-controller e2e suite - 6/312 specs SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
• [SLOW TEST:33.501 seconds]
[Serial] admission controller should return an error if the Ingress V1 definition contains invalid annotations
/go/src/k8s.io/ingress-nginx/test/e2e/admission/admission.go:143
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
• [SLOW TEST:32.811 seconds]
[Serial] admission controller reject ingress with global-rate-limit annotations when memcached is not configured
/go/src/k8s.io/ingress-nginx/test/e2e/admission/admission.go:48
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
• [SLOW TEST:26.040 seconds]
[Serial] admission controller should allow overlaps of host and paths with canary annotation
/go/src/k8s.io/ingress-nginx/test/e2e/admission/admission.go:92
------------------------------
SSSSSSS
------------------------------
• [SLOW TEST:32.011 seconds]
[Serial] admission controller should not allow overlaps of host and paths without canary annotations
/go/src/k8s.io/ingress-nginx/test/e2e/admission/admission.go:75
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
• [SLOW TEST:33.306 seconds]
[Serial] admission controller should not return an error if the Ingress V1 definition is valid
/go/src/k8s.io/ingress-nginx/test/e2e/admission/admission.go:124
------------------------------
SSSSSSSSSSSSSSSS
------------------------------
• [SLOW TEST:28.871 seconds]
[Serial] admission controller should return an error if there is an error validating the ingress definition
/go/src/k8s.io/ingress-nginx/test/e2e/admission/admission.go:113
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS SUCCESS! 3m6.5421336s PASS

Ginkgo ran 1 suite in 3m6.5953325s
Test Suite Passed
pod "e2e" deleted
Deleting cluster "ingress-nginx-dev" ...
bouke@***:/mnt/h/temp/ingress-nginx$
```

</details>